<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Data Processing | Sequencing databases in Shiny</title>
  <meta name="description" content="Here we will describe how to build a Shiny app for sequencing data using the code base described in CITE PAPER HERE." />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Data Processing | Sequencing databases in Shiny" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Here we will describe how to build a Shiny app for sequencing data using the code base described in CITE PAPER HERE." />
  <meta name="github-repo" content="aliibarry/omics-database" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Data Processing | Sequencing databases in Shiny" />
  
  <meta name="twitter:description" content="Here we will describe how to build a Shiny app for sequencing data using the code base described in CITE PAPER HERE." />
  

<meta name="author" content="AMB and NZ" />


<meta name="date" content="2023-03-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="ui.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Sequencing databases in Shiny</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#shiny"><i class="fa fa-check"></i><b>1.1</b> Shiny</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow"><i class="fa fa-check"></i><b>1.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>2</b> Data Processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data.html"><a href="data.html#preparation"><i class="fa fa-check"></i><b>2.1</b> Preparation</a></li>
<li class="chapter" data-level="2.2" data-path="data.html"><a href="data.html#species-optional"><i class="fa fa-check"></i><b>2.2</b> Species <em>(optional)</em></a></li>
<li class="chapter" data-level="2.3" data-path="data.html"><a href="data.html#rdata-option-1"><i class="fa fa-check"></i><b>2.3</b> RData <em>(option 1)</em></a></li>
<li class="chapter" data-level="2.4" data-path="data.html"><a href="data.html#building-an-sql-database-via-r-recommended"><i class="fa fa-check"></i><b>2.4</b> Building an SQL database via R <em>(recommended)</em></a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="data.html"><a href="data.html#sql-manipulation"><i class="fa fa-check"></i><b>2.4.1</b> SQL manipulation</a></li>
<li class="chapter" data-level="2.4.2" data-path="data.html"><a href="data.html#processing-multiple-files"><i class="fa fa-check"></i><b>2.4.2</b> Processing multiple files</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="ui.html"><a href="ui.html"><i class="fa fa-check"></i><b>3</b> User Interface</a>
<ul>
<li class="chapter" data-level="3.1" data-path="ui.html"><a href="ui.html#basic-structure"><i class="fa fa-check"></i><b>3.1</b> Basic structure</a></li>
<li class="chapter" data-level="3.2" data-path="ui.html"><a href="ui.html#header"><i class="fa fa-check"></i><b>3.2</b> Header</a></li>
<li class="chapter" data-level="3.3" data-path="ui.html"><a href="ui.html#sidebar"><i class="fa fa-check"></i><b>3.3</b> Sidebar</a></li>
<li class="chapter" data-level="3.4" data-path="ui.html"><a href="ui.html#body"><i class="fa fa-check"></i><b>3.4</b> Body</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="ui.html"><a href="ui.html#homepage"><i class="fa fa-check"></i><b>3.4.1</b> Homepage</a></li>
<li class="chapter" data-level="3.4.2" data-path="ui.html"><a href="ui.html#grouped-comparisons"><i class="fa fa-check"></i><b>3.4.2</b> Grouped comparisons</a></li>
<li class="chapter" data-level="3.4.3" data-path="ui.html"><a href="ui.html#individual-analysis"><i class="fa fa-check"></i><b>3.4.3</b> Individual Analysis</a></li>
<li class="chapter" data-level="3.4.4" data-path="ui.html"><a href="ui.html#network-page"><i class="fa fa-check"></i><b>3.4.4</b> Network page</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="server.html"><a href="server.html"><i class="fa fa-check"></i><b>4</b> Server</a>
<ul>
<li class="chapter" data-level="4.1" data-path="server.html"><a href="server.html#requirements"><i class="fa fa-check"></i><b>4.1</b> Requirements</a></li>
<li class="chapter" data-level="4.2" data-path="server.html"><a href="server.html#server-overview"><i class="fa fa-check"></i><b>4.2</b> Server Overview</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="server.html"><a href="server.html#side-bar"><i class="fa fa-check"></i><b>4.2.1</b> Side bar</a></li>
<li class="chapter" data-level="4.2.2" data-path="server.html"><a href="server.html#tab-menu"><i class="fa fa-check"></i><b>4.2.2</b> Tab Menu</a></li>
<li class="chapter" data-level="4.2.3" data-path="server.html"><a href="server.html#grouped-comparisons-1"><i class="fa fa-check"></i><b>4.2.3</b> Grouped comparisons</a></li>
<li class="chapter" data-level="4.2.4" data-path="server.html"><a href="server.html#individual-analysis-1"><i class="fa fa-check"></i><b>4.2.4</b> Individual Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="server.html"><a href="server.html#network-analysis"><i class="fa fa-check"></i><b>4.3</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="server.html"><a href="server.html#integration-with-string-db"><i class="fa fa-check"></i><b>4.3.1</b> Integration with STRING DB</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="server.html"><a href="server.html#functions"><i class="fa fa-check"></i><b>4.4</b> Functions</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="server.html"><a href="server.html#data-processing"><i class="fa fa-check"></i><b>4.4.1</b> Data processing</a></li>
<li class="chapter" data-level="4.4.2" data-path="server.html"><a href="server.html#dot-plots"><i class="fa fa-check"></i><b>4.4.2</b> Dot plots</a></li>
<li class="chapter" data-level="4.4.3" data-path="server.html"><a href="server.html#line-plots"><i class="fa fa-check"></i><b>4.4.3</b> Line plots</a></li>
<li class="chapter" data-level="4.4.4" data-path="server.html"><a href="server.html#a-differential-expression-dot-plot"><i class="fa fa-check"></i><b>4.4.4</b> A differential expression dot plot</a></li>
<li class="chapter" data-level="4.4.5" data-path="server.html"><a href="server.html#a-volcano-plot"><i class="fa fa-check"></i><b>4.4.5</b> A volcano plot</a></li>
<li class="chapter" data-level="4.4.6" data-path="server.html"><a href="server.html#other"><i class="fa fa-check"></i><b>4.4.6</b> Other</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="deploy.html"><a href="deploy.html"><i class="fa fa-check"></i><b>5</b> Deploying your application</a>
<ul>
<li class="chapter" data-level="5.1" data-path="deploy.html"><a href="deploy.html#shinyapps.io"><i class="fa fa-check"></i><b>5.1</b> Shinyapps.io</a></li>
<li class="chapter" data-level="5.2" data-path="deploy.html"><a href="deploy.html#bioconductor"><i class="fa fa-check"></i><b>5.2</b> Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="trouble.html"><a href="trouble.html"><i class="fa fa-check"></i><b>6</b> Troubleshooting</a>
<ul>
<li class="chapter" data-level="6.1" data-path="trouble.html"><a href="trouble.html#package-installation"><i class="fa fa-check"></i><b>6.1</b> Package installation</a></li>
<li class="chapter" data-level="6.2" data-path="trouble.html"><a href="trouble.html#file-setup"><i class="fa fa-check"></i><b>6.2</b> File setup</a></li>
<li class="chapter" data-level="6.3" data-path="trouble.html"><a href="trouble.html#styling"><i class="fa fa-check"></i><b>6.3</b> Styling</a></li>
<li class="chapter" data-level="6.4" data-path="trouble.html"><a href="trouble.html#data-inputoutput-issues"><i class="fa fa-check"></i><b>6.4</b> Data input/output issues</a></li>
<li class="chapter" data-level="6.5" data-path="trouble.html"><a href="trouble.html#debugging-reactive-code"><i class="fa fa-check"></i><b>6.5</b> Debugging reactive code</a></li>
<li class="chapter" data-level="6.6" data-path="trouble.html"><a href="trouble.html#deployment"><i class="fa fa-check"></i><b>6.6</b> Deployment</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="trouble.html"><a href="trouble.html#basics"><i class="fa fa-check"></i><b>6.6.1</b> Basics</a></li>
<li class="chapter" data-level="6.6.2" data-path="trouble.html"><a href="trouble.html#bioconductor-1"><i class="fa fa-check"></i><b>6.6.2</b> Bioconductor</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Sequencing databases in Shiny</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Data Processing<a href="data.html#data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<style>
img
{
    display:block;
    float:none;
    margin-left:auto;
    margin-right:auto;
}
</style>
<ol style="list-style-type: decimal">
<li>Prepare data tables (minimum: count matrix, column data)</li>
<li>Compile all data into an <code>.RData</code> or <code>.db</code> object.</li>
</ol>
<hr />
<div id="preparation" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Preparation<a href="data.html#preparation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To implement this app, you will need three types of data:</p>
<ol style="list-style-type: decimal">
<li>Count tables in <code>./data/CountTables/</code> folder</li>
</ol>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:250px; overflow-x: scroll; width:100%; ">
<table class="table table-striped table-hover table-condensed table-responsive" style="font-size: 12px; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
X
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample50_BALB.c_SHAM_M
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample51_BALB.c_SHAM_M
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample55_BALB.c_SHAM_F
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample56_BALB.c_SHAM_F
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample57_BALB.c_SHAM_F
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample58_B10.D2_SHAM_M
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample60_B10.D2_SHAM_M
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample67_B10.D2_SHAM_F
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample71_BALB.c_SHAM_M
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample77_B10.D2_SHAM_F
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample88_B10.D2_SHAM_M
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample49_BALB.c_SNI_M
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample64_BALB.c_SNI_F
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample65_BALB.c_SNI_F
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample69_B10.D2_SNI_F
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample70_BALB.c_SNI_M
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample76_B10.D2_SNI_F
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample78_B10.D2_SNI_F
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample89_B10.D2_SNI_M
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Sample90_B10.D2_SNI_M
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
symbol
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ENSMUSG00000000001
</td>
<td style="text-align:right;">
20.6460082
</td>
<td style="text-align:right;">
20.9684737
</td>
<td style="text-align:right;">
25.6051105
</td>
<td style="text-align:right;">
17.9245459
</td>
<td style="text-align:right;">
21.8279448
</td>
<td style="text-align:right;">
23.8898091
</td>
<td style="text-align:right;">
20.8612482
</td>
<td style="text-align:right;">
21.1079774
</td>
<td style="text-align:right;">
21.9353412
</td>
<td style="text-align:right;">
19.8810344
</td>
<td style="text-align:right;">
25.4903826
</td>
<td style="text-align:right;">
23.2414339
</td>
<td style="text-align:right;">
26.0955758
</td>
<td style="text-align:right;">
24.5983920
</td>
<td style="text-align:right;">
22.3698430
</td>
<td style="text-align:right;">
25.4514816
</td>
<td style="text-align:right;">
23.2571636
</td>
<td style="text-align:right;">
25.0393354
</td>
<td style="text-align:right;">
21.4696758
</td>
<td style="text-align:right;">
21.4316521
</td>
<td style="text-align:left;">
Gnai3
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSMUSG00000000003
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:left;">
Pbsn
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSMUSG00000000028
</td>
<td style="text-align:right;">
1.2444692
</td>
<td style="text-align:right;">
1.0386601
</td>
<td style="text-align:right;">
1.4225152
</td>
<td style="text-align:right;">
0.8771354
</td>
<td style="text-align:right;">
0.7774382
</td>
<td style="text-align:right;">
0.9572740
</td>
<td style="text-align:right;">
0.6793672
</td>
<td style="text-align:right;">
0.8532159
</td>
<td style="text-align:right;">
0.9904448
</td>
<td style="text-align:right;">
0.8884526
</td>
<td style="text-align:right;">
1.1943520
</td>
<td style="text-align:right;">
1.4496099
</td>
<td style="text-align:right;">
1.0502148
</td>
<td style="text-align:right;">
0.9892706
</td>
<td style="text-align:right;">
0.8066116
</td>
<td style="text-align:right;">
0.8839994
</td>
<td style="text-align:right;">
0.9460189
</td>
<td style="text-align:right;">
0.9714055
</td>
<td style="text-align:right;">
0.9352520
</td>
<td style="text-align:right;">
0.9411879
</td>
<td style="text-align:left;">
Cdc45
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSMUSG00000000031
</td>
<td style="text-align:right;">
146.6304065
</td>
<td style="text-align:right;">
78.6870749
</td>
<td style="text-align:right;">
279.4541932
</td>
<td style="text-align:right;">
122.2226711
</td>
<td style="text-align:right;">
86.5995137
</td>
<td style="text-align:right;">
385.7192026
</td>
<td style="text-align:right;">
402.3015644
</td>
<td style="text-align:right;">
373.7624763
</td>
<td style="text-align:right;">
83.4889245
</td>
<td style="text-align:right;">
416.2586323
</td>
<td style="text-align:right;">
523.7706518
</td>
<td style="text-align:right;">
34.6120591
</td>
<td style="text-align:right;">
215.3297763
</td>
<td style="text-align:right;">
259.9150108
</td>
<td style="text-align:right;">
642.1622405
</td>
<td style="text-align:right;">
103.1170442
</td>
<td style="text-align:right;">
510.7111982
</td>
<td style="text-align:right;">
497.6381737
</td>
<td style="text-align:right;">
213.8979427
</td>
<td style="text-align:right;">
462.2669044
</td>
<td style="text-align:left;">
H19
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSMUSG00000000037
</td>
<td style="text-align:right;">
0.0673974
</td>
<td style="text-align:right;">
0.0434038
</td>
<td style="text-align:right;">
0.0736720
</td>
<td style="text-align:right;">
0.0567710
</td>
<td style="text-align:right;">
0.0603054
</td>
<td style="text-align:right;">
0.0782123
</td>
<td style="text-align:right;">
0.0638765
</td>
<td style="text-align:right;">
0.0663909
</td>
<td style="text-align:right;">
0.0661561
</td>
<td style="text-align:right;">
0.0811122
</td>
<td style="text-align:right;">
0.1074937
</td>
<td style="text-align:right;">
0.0509788
</td>
<td style="text-align:right;">
0.0717639
</td>
<td style="text-align:right;">
0.0585030
</td>
<td style="text-align:right;">
0.0728068
</td>
<td style="text-align:right;">
0.0970444
</td>
<td style="text-align:right;">
0.0831925
</td>
<td style="text-align:right;">
0.1111677
</td>
<td style="text-align:right;">
0.0729349
</td>
<td style="text-align:right;">
0.0880621
</td>
<td style="text-align:left;">
Scml2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSMUSG00000000049
</td>
<td style="text-align:right;">
0.0895870
</td>
<td style="text-align:right;">
0.0961564
</td>
<td style="text-align:right;">
0.0699482
</td>
<td style="text-align:right;">
0.0611854
</td>
<td style="text-align:right;">
0.1214547
</td>
<td style="text-align:right;">
0.1831722
</td>
<td style="text-align:right;">
0.1309992
</td>
<td style="text-align:right;">
0.1764983
</td>
<td style="text-align:right;">
0.0713003
</td>
<td style="text-align:right;">
0.1952917
</td>
<td style="text-align:right;">
0.2609194
</td>
<td style="text-align:right;">
0.1016442
</td>
<td style="text-align:right;">
0.0931727
</td>
<td style="text-align:right;">
0.0818572
</td>
<td style="text-align:right;">
0.1539641
</td>
<td style="text-align:right;">
0.1098845
</td>
<td style="text-align:right;">
0.2628942
</td>
<td style="text-align:right;">
0.1581742
</td>
<td style="text-align:right;">
0.1856444
</td>
<td style="text-align:right;">
0.2387010
</td>
<td style="text-align:left;">
Apoh
</td>
</tr>
</tbody>
</table>
</div>
<p> </p>
<ol start="2" style="list-style-type: decimal">
<li>Column Data (abbrev: <code>colData</code>) in <code>./data/ColDatas/</code> folder</li>
</ol>
<ul>
<li>This contains the info about each sample</li>
<li>The colnames of <code>colData</code> should include <code>Population</code>, <code>Timepoint</code>, <code>Sex</code>, <code>Dataset</code>, and <code>Species</code></li>
<li>If your experiment does not separate those samples based on any of those parameters, you will need to create a column that specifies this. E.g. if your samples are of mixed sex, you can creates a <code>Sex</code> column containing ‘mixed’ for all samples.</li>
</ul>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:100%; ">
<table class="table table-striped table-hover table-condensed table-responsive" style="font-size: 12px; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
X
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Sample
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Population
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Condition
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Sex
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Sample_Pop_Cond_Sex
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Dataset
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Species
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Timepoint
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Sample50_BALB.c_SHAM_M
</td>
<td style="text-align:left;">
Sample50
</td>
<td style="text-align:left;">
BALB.c
</td>
<td style="text-align:left;">
SHAM
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
Sample50_BALB.c_SHAM_M
</td>
<td style="text-align:left;">
Bulk DRG
</td>
<td style="text-align:left;">
Mouse (DRG)
</td>
<td style="text-align:left;">
na
</td>
</tr>
<tr>
<td style="text-align:left;">
Sample51_BALB.c_SHAM_M
</td>
<td style="text-align:left;">
Sample51
</td>
<td style="text-align:left;">
BALB.c
</td>
<td style="text-align:left;">
SHAM
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
Sample51_BALB.c_SHAM_M
</td>
<td style="text-align:left;">
Bulk DRG
</td>
<td style="text-align:left;">
Mouse (DRG)
</td>
<td style="text-align:left;">
na
</td>
</tr>
<tr>
<td style="text-align:left;">
Sample55_BALB.c_SHAM_F
</td>
<td style="text-align:left;">
Sample55
</td>
<td style="text-align:left;">
BALB.c
</td>
<td style="text-align:left;">
SHAM
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
Sample55_BALB.c_SHAM_F
</td>
<td style="text-align:left;">
Bulk DRG
</td>
<td style="text-align:left;">
Mouse (DRG)
</td>
<td style="text-align:left;">
na
</td>
</tr>
<tr>
<td style="text-align:left;">
Sample56_BALB.c_SHAM_F
</td>
<td style="text-align:left;">
Sample56
</td>
<td style="text-align:left;">
BALB.c
</td>
<td style="text-align:left;">
SHAM
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
Sample56_BALB.c_SHAM_F
</td>
<td style="text-align:left;">
Bulk DRG
</td>
<td style="text-align:left;">
Mouse (DRG)
</td>
<td style="text-align:left;">
na
</td>
</tr>
<tr>
<td style="text-align:left;">
Sample57_BALB.c_SHAM_F
</td>
<td style="text-align:left;">
Sample57
</td>
<td style="text-align:left;">
BALB.c
</td>
<td style="text-align:left;">
SHAM
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
Sample57_BALB.c_SHAM_F
</td>
<td style="text-align:left;">
Bulk DRG
</td>
<td style="text-align:left;">
Mouse (DRG)
</td>
<td style="text-align:left;">
na
</td>
</tr>
<tr>
<td style="text-align:left;">
Sample58_B10.D2_SHAM_M
</td>
<td style="text-align:left;">
Sample58
</td>
<td style="text-align:left;">
B10.D2
</td>
<td style="text-align:left;">
SHAM
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
Sample58_B10.D2_SHAM_M
</td>
<td style="text-align:left;">
Bulk DRG
</td>
<td style="text-align:left;">
Mouse (DRG)
</td>
<td style="text-align:left;">
na
</td>
</tr>
</tbody>
</table>
</div>
<p> </p>
<ol start="3" style="list-style-type: decimal">
<li>Differential expression data frames (abbrev: <code>deg_df</code>) in <code>./data/DegData/</code> folder<br />
</li>
</ol>
<ul>
<li>This data file contains the <em>log2FoldChange</em> and statistics of one case vs control</li>
<li>An experiment can have multiple <code>deg_df</code> files</li>
</ul>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:250px; overflow-x: scroll; width:100%; ">
<table class="table table-striped table-hover table-condensed table-responsive" style="font-size: 12px; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
baseMean
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
log2FoldChange
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
lfcSE
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
stat
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
pvalue
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
padj
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
symbol
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Population
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ENSMUSG00000109908
</td>
<td style="text-align:right;">
10.737773
</td>
<td style="text-align:right;">
6.453185
</td>
<td style="text-align:right;">
4.103313
</td>
<td style="text-align:right;">
1.572677
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
b10d2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSMUSG00000096108
</td>
<td style="text-align:right;">
4.372395
</td>
<td style="text-align:right;">
-6.405612
</td>
<td style="text-align:right;">
3.800576
</td>
<td style="text-align:right;">
-1.685432
</td>
<td style="text-align:right;">
0.0919053
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
b10d2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSMUSG00000095889
</td>
<td style="text-align:right;">
3.541538
</td>
<td style="text-align:right;">
-6.168911
</td>
<td style="text-align:right;">
4.228860
</td>
<td style="text-align:right;">
-1.458765
</td>
<td style="text-align:right;">
0.1446299
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
b10d2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSMUSG00000073494
</td>
<td style="text-align:right;">
5.130631
</td>
<td style="text-align:right;">
5.941862
</td>
<td style="text-align:right;">
1.068377
</td>
<td style="text-align:right;">
5.561578
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
Sh2d1b2
</td>
<td style="text-align:left;">
b10d2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSMUSG00000036357
</td>
<td style="text-align:right;">
3.708027
</td>
<td style="text-align:right;">
5.483357
</td>
<td style="text-align:right;">
1.130411
</td>
<td style="text-align:right;">
4.850764
</td>
<td style="text-align:right;">
0.0000012
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
Gpr101
</td>
<td style="text-align:left;">
b10d2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSMUSG00000076538
</td>
<td style="text-align:right;">
3.050185
</td>
<td style="text-align:right;">
-5.384935
</td>
<td style="text-align:right;">
1.643832
</td>
<td style="text-align:right;">
-3.275843
</td>
<td style="text-align:right;">
0.0010535
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
b10d2
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="species-optional" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Species <em>(optional)</em><a href="data.html#species-optional" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If your data contains different species, you will also need datasets that allow you map gene symbols/IDs between species. We have provided an option using the <code>biomaRt</code> library.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="data.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;biomaRt&quot;</span>)</span>
<span id="cb1-2"><a href="data.html#cb1-2" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">=</span> <span class="fu">useMart</span>(<span class="st">&quot;ensembl&quot;</span>, <span class="at">dataset =</span> <span class="st">&quot;mmusculus_gene_ensembl&quot;</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>, <span class="at">host =</span> <span class="st">&quot;https://dec2021.archive.ensembl.org&quot;</span>)</span>
<span id="cb1-3"><a href="data.html#cb1-3" aria-hidden="true" tabindex="-1"></a>human <span class="ot">=</span> <span class="fu">useMart</span>(<span class="st">&quot;ensembl&quot;</span>, <span class="at">dataset =</span> <span class="st">&quot;hsapiens_gene_ensembl&quot;</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>, <span class="at">host =</span> <span class="st">&quot;https://dec2021.archive.ensembl.org&quot;</span>)</span>
<span id="cb1-4"><a href="data.html#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="data.html#cb1-5" aria-hidden="true" tabindex="-1"></a>geneV1 <span class="ot">=</span> <span class="fu">getLDS</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">&quot;mgi_symbol&quot;</span>,<span class="st">&quot;ensembl_gene_id&quot;</span>), </span>
<span id="cb1-6"><a href="data.html#cb1-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">filters =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>, </span>
<span id="cb1-7"><a href="data.html#cb1-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">values =</span> count<span class="sc">$</span>X, </span>
<span id="cb1-8"><a href="data.html#cb1-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">mart =</span> mouse, </span>
<span id="cb1-9"><a href="data.html#cb1-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">attributesL =</span> <span class="fu">c</span>(<span class="st">&quot;hgnc_symbol&quot;</span>, <span class="st">&quot;ensembl_gene_id&quot;</span>, <span class="st">&quot;description&quot;</span>), </span>
<span id="cb1-10"><a href="data.html#cb1-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">martL =</span> human, </span>
<span id="cb1-11"><a href="data.html#cb1-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">uniqueRows=</span>T)</span>
<span id="cb1-12"><a href="data.html#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="data.html#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(geneV1) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;mgi_symbol&quot;</span>, <span class="st">&quot;mouse_gene_id&quot;</span>, <span class="st">&quot;hgnc_symbol&quot;</span>, <span class="st">&quot;human_gene_id&quot;</span>)</span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:450px; overflow-x: scroll; width:100%; ">
<table class="table table-striped table-hover table-condensed table-responsive" style="font-size: 12px; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
mgi_symbol
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
mouse_gene_id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
hgnc_symbol
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
human_gene_id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
NA
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
mt-Nd5
</td>
<td style="text-align:left;">
ENSMUSG00000064367
</td>
<td style="text-align:left;">
MT-ND5
</td>
<td style="text-align:left;">
ENSG00000198786
</td>
<td style="text-align:left;">
mitochondrially encoded NADH:ubiquinone oxidoreductase core subunit 5 [Source:HGNC Symbol;Acc:HGNC:7461]
</td>
</tr>
<tr>
<td style="text-align:left;">
mt-Co2
</td>
<td style="text-align:left;">
ENSMUSG00000064354
</td>
<td style="text-align:left;">
MT-CO2
</td>
<td style="text-align:left;">
ENSG00000198712
</td>
<td style="text-align:left;">
mitochondrially encoded cytochrome c oxidase II [Source:HGNC Symbol;Acc:HGNC:7421]
</td>
</tr>
<tr>
<td style="text-align:left;">
mt-Co3
</td>
<td style="text-align:left;">
ENSMUSG00000064358
</td>
<td style="text-align:left;">
MT-CO3
</td>
<td style="text-align:left;">
ENSG00000198938
</td>
<td style="text-align:left;">
mitochondrially encoded cytochrome c oxidase III [Source:HGNC Symbol;Acc:HGNC:7422]
</td>
</tr>
<tr>
<td style="text-align:left;">
mt-Cytb
</td>
<td style="text-align:left;">
ENSMUSG00000064370
</td>
<td style="text-align:left;">
MT-CYB
</td>
<td style="text-align:left;">
ENSG00000198727
</td>
<td style="text-align:left;">
mitochondrially encoded cytochrome b [Source:HGNC Symbol;Acc:HGNC:7427]
</td>
</tr>
<tr>
<td style="text-align:left;">
mt-Nd4l
</td>
<td style="text-align:left;">
ENSMUSG00000065947
</td>
<td style="text-align:left;">
MT-ND4L
</td>
<td style="text-align:left;">
ENSG00000212907
</td>
<td style="text-align:left;">
mitochondrially encoded NADH:ubiquinone oxidoreductase core subunit 4L [Source:HGNC Symbol;Acc:HGNC:7460]
</td>
</tr>
<tr>
<td style="text-align:left;">
mt-Nd2
</td>
<td style="text-align:left;">
ENSMUSG00000064345
</td>
<td style="text-align:left;">
MT-ND2
</td>
<td style="text-align:left;">
ENSG00000198763
</td>
<td style="text-align:left;">
mitochondrially encoded NADH:ubiquinone oxidoreductase core subunit 2 [Source:HGNC Symbol;Acc:HGNC:7456]
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="rdata-option-1" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> RData <em>(option 1)</em><a href="data.html#rdata-option-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You can store data frames into .RData using the <code>save()</code> function. For example, we can use the save() function to save the <code>count</code> data frame to a file called <code>my_data.RData</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="data.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(count, <span class="at">file =</span> <span class="st">&quot;my_data.RData&quot;</span>)</span></code></pre></div>
<p>To load the data frame from the <code>.RData</code> file later, you can use the <code>load()</code> function in your server. Y</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="data.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;my_data.RData&quot;</span>) <span class="co"># add to server</span></span></code></pre></div>
<p><em>NOTE</em>, some functions will need to be modified to read from a <code>.RData</code> instead. This can be done by uncommentng the relevant lines of code. For example:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="data.html#cb4-1" aria-hidden="true" tabindex="-1"></a>  shiny<span class="sc">::</span><span class="fu">observeEvent</span>(input<span class="sc">$</span>load, {</span>
<span id="cb4-2"><a href="data.html#cb4-2" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-3"><a href="data.html#cb4-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (include_count <span class="sc">==</span> <span class="cn">TRUE</span>){</span>
<span id="cb4-4"><a href="data.html#cb4-4" aria-hidden="true" tabindex="-1"></a>      <span class="co"># query count datasets</span></span>
<span id="cb4-5"><a href="data.html#cb4-5" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb4-6"><a href="data.html#cb4-6" aria-hidden="true" tabindex="-1"></a>      <span class="do">## comment here for .RData file</span></span>
<span id="cb4-7"><a href="data.html#cb4-7" aria-hidden="true" tabindex="-1"></a>      <span class="co"># for database </span></span>
<span id="cb4-8"><a href="data.html#cb4-8" aria-hidden="true" tabindex="-1"></a>      {</span>
<span id="cb4-9"><a href="data.html#cb4-9" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (species <span class="sc">==</span> <span class="st">&quot;human&quot;</span>) {</span>
<span id="cb4-10"><a href="data.html#cb4-10" aria-hidden="true" tabindex="-1"></a>          sql <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;SELECT * FROM&quot;</span>, count_df, <span class="st">&quot;WHERE mgi_symbol = ?&quot;</span>)</span>
<span id="cb4-11"><a href="data.html#cb4-11" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb4-12"><a href="data.html#cb4-12" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>{</span>
<span id="cb4-13"><a href="data.html#cb4-13" aria-hidden="true" tabindex="-1"></a>          sql <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;SELECT * FROM&quot;</span>, count_df, <span class="st">&quot;WHERE symbol = ?&quot;</span>)</span>
<span id="cb4-14"><a href="data.html#cb4-14" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb4-15"><a href="data.html#cb4-15" aria-hidden="true" tabindex="-1"></a>        filt <span class="ot">=</span> RSQLite<span class="sc">::</span><span class="fu">dbGetPreparedQuery</span>(conn, sql, <span class="at">bind.data=</span><span class="fu">data.frame</span>(<span class="at">symbol=</span><span class="fu">genes</span>()))</span>
<span id="cb4-16"><a href="data.html#cb4-16" aria-hidden="true" tabindex="-1"></a>        filt <span class="ot">=</span> <span class="fu">as.data.frame</span>(filt)</span>
<span id="cb4-17"><a href="data.html#cb4-17" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb4-18"><a href="data.html#cb4-18" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb4-19"><a href="data.html#cb4-19" aria-hidden="true" tabindex="-1"></a>      <span class="do">## uncomment here for .RData file</span></span>
<span id="cb4-20"><a href="data.html#cb4-20" aria-hidden="true" tabindex="-1"></a>      <span class="co"># using R Data </span></span>
<span id="cb4-21"><a href="data.html#cb4-21" aria-hidden="true" tabindex="-1"></a>      {</span>
<span id="cb4-22"><a href="data.html#cb4-22" aria-hidden="true" tabindex="-1"></a>        <span class="co"># if (species == &quot;human&quot;){</span></span>
<span id="cb4-23"><a href="data.html#cb4-23" aria-hidden="true" tabindex="-1"></a>        <span class="co">#   filt = count_df[count_df$mgi_symbol %in% genes(),]</span></span>
<span id="cb4-24"><a href="data.html#cb4-24" aria-hidden="true" tabindex="-1"></a>        <span class="co">#   X = rownames(filt)</span></span>
<span id="cb4-25"><a href="data.html#cb4-25" aria-hidden="true" tabindex="-1"></a>        <span class="co">#   filt = cbind(X, filt)</span></span>
<span id="cb4-26"><a href="data.html#cb4-26" aria-hidden="true" tabindex="-1"></a>        <span class="co">#   </span></span>
<span id="cb4-27"><a href="data.html#cb4-27" aria-hidden="true" tabindex="-1"></a>        <span class="co"># }</span></span>
<span id="cb4-28"><a href="data.html#cb4-28" aria-hidden="true" tabindex="-1"></a>        <span class="co"># else {</span></span>
<span id="cb4-29"><a href="data.html#cb4-29" aria-hidden="true" tabindex="-1"></a>        <span class="co">#   filt = count_df[count_df$symbol %in% genes(),]</span></span>
<span id="cb4-30"><a href="data.html#cb4-30" aria-hidden="true" tabindex="-1"></a>        <span class="co">#   X = rownames(filt)</span></span>
<span id="cb4-31"><a href="data.html#cb4-31" aria-hidden="true" tabindex="-1"></a>        <span class="co">#   filt = cbind(X, filt)</span></span>
<span id="cb4-32"><a href="data.html#cb4-32" aria-hidden="true" tabindex="-1"></a>        <span class="co"># }</span></span>
<span id="cb4-33"><a href="data.html#cb4-33" aria-hidden="true" tabindex="-1"></a>      }</span></code></pre></div>
<p>If you plan to store data this way, move to the <em>User Interface</em> section now. If you are working with larger amounts of data, you may find it useful to set up an SQL database (<em>recommended</em>, below).</p>
</div>
<div id="building-an-sql-database-via-r-recommended" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Building an SQL database via R <em>(recommended)</em><a href="data.html#building-an-sql-database-via-r-recommended" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>While <code>.RData</code> files can be useful small or medium-sized datasets in R, SQL databases provide a more powerful and flexible solution for managing large, complex datasets that need to be shared, queried, and analyzed by multiple users or integrated with other tools and applications.</p>
<p>An SQLite database consists of one or more tables, which are composed of rows and columns. An RSQLite <code>.db</code> object is a database connection to a SQLite database file through R. The following functions will be useful in creating and updating the <code>.db</code> object in R.</p>
<p>Here, we provide an overview for bulk count data. <code>Seurat</code> objects (for sc/snRNA-seq or ATAC-seq) are read in separately.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="data.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This function connects to a SQLite database file called `example_database.db`</span></span>
<span id="cb5-2"><a href="data.html#cb5-2" aria-hidden="true" tabindex="-1"></a>db_dir <span class="ot">=</span> <span class="st">&quot;example_database.db&quot;</span> </span>
<span id="cb5-3"><a href="data.html#cb5-3" aria-hidden="true" tabindex="-1"></a>conn  <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), db_dir)</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="data.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This function stores dataframe objects into `.db` object </span></span>
<span id="cb6-2"><a href="data.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dbWriteTable</span>(conn, <span class="at">name =</span> <span class="st">&quot;count_df&quot;</span>, <span class="at">value =</span> count)</span></code></pre></div>
<div id="sql-manipulation" class="section level3 hasAnchor" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> SQL manipulation<a href="data.html#sql-manipulation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="data.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># List table names that you have already stored  </span></span>
<span id="cb7-2"><a href="data.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dbListTables</span>(conn)</span></code></pre></div>
<pre><code>## [1] &quot;count_df&quot;</code></pre>
<p>Each table in a SQLite database has a schema that defines the columns in the table and their data types. In R, you can retrieve the schema of a table using the <code>dbListFields()</code> function, which returns a character vector of the column names.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="data.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the schema of the &quot;count_df&quot; table</span></span>
<span id="cb9-2"><a href="data.html#cb9-2" aria-hidden="true" tabindex="-1"></a>fields <span class="ot">&lt;-</span> <span class="fu">dbListFields</span>(conn, <span class="st">&quot;count_df&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="data.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Modify a table by using &#39;overwrite&#39; argument </span></span>
<span id="cb10-2"><a href="data.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dbWriteTable</span>(conn, <span class="at">name =</span> <span class="st">&quot;count_df&quot;</span>, <span class="at">value =</span> count_modified, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="data.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Delete a table </span></span>
<span id="cb11-2"><a href="data.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dbRemoveTable</span>(conn, <span class="st">&quot;count_df&quot;</span>)</span></code></pre></div>
<p>The <code>dbGetQuery()</code> function in RSQLite allows you to execute SQL queries and return the results as a data frame.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="data.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select all rows from the &quot;count_df&quot; table</span></span>
<span id="cb12-2"><a href="data.html#cb12-2" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">&quot;SELECT * FROM count_df&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="data.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the connection after you stored all tables</span></span>
<span id="cb13-2"><a href="data.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dbDisconnect</span>(conn)</span></code></pre></div>
<p>These functions are sufficient for users without prior knowledge in the RSQLite package. Further tutorials can be found at <a href="https://www.datacamp.com/tutorial/sqlite-in-r" class="uri">https://www.datacamp.com/tutorial/sqlite-in-r</a></p>
</div>
<div id="processing-multiple-files" class="section level3 hasAnchor" number="2.4.2">
<h3><span class="header-section-number">2.4.2</span> Processing multiple files<a href="data.html#processing-multiple-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If your experimental data contains multiple files, it can become labor intensive to write tables one by one. To reduce the burden, you can read files by folders.</p>
<p>• Read all count tables in <code>./data/CountTables</code> folder into <code>.db</code> object, using their file names as table names.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="data.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># all file names </span></span>
<span id="cb14-2"><a href="data.html#cb14-2" aria-hidden="true" tabindex="-1"></a>Counts <span class="ot">&lt;-</span> <span class="fu">str_sort</span>(<span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">&quot;./data/CountTables&quot;</span>, <span class="at">pattern =</span> <span class="st">&quot;*.csv&quot;</span>)) <span class="co"># names of the count tables</span></span>
<span id="cb14-3"><a href="data.html#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="data.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># name each file as their file name without &quot;.csv&quot; suffix </span></span>
<span id="cb14-5"><a href="data.html#cb14-5" aria-hidden="true" tabindex="-1"></a>names <span class="ot">=</span> <span class="fu">as.vector</span>(<span class="fu">substr</span>(Counts, <span class="dv">0</span>, <span class="fu">nchar</span>(Counts) <span class="sc">-</span> <span class="dv">4</span>))</span>
<span id="cb14-6"><a href="data.html#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="data.html#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co"># read files into database, using their file names as table names </span></span>
<span id="cb14-8"><a href="data.html#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(names)){</span>
<span id="cb14-9"><a href="data.html#cb14-9" aria-hidden="true" tabindex="-1"></a>  name  <span class="ot">=</span> names[i]</span>
<span id="cb14-10"><a href="data.html#cb14-10" aria-hidden="true" tabindex="-1"></a>  value <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(<span class="st">&quot;./data/CountTables/&quot;</span>, Counts[i]), <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-11"><a href="data.html#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dbWriteTable</span>(conn, name, value)</span>
<span id="cb14-12"><a href="data.html#cb14-12" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>• Read ColDatas files in <code>./data/ColDatas</code> directory into tables.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="data.html#cb15-1" aria-hidden="true" tabindex="-1"></a>CDs <span class="ot">&lt;-</span> <span class="fu">str_sort</span>(<span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">&quot;./data/ColDatas&quot;</span>, <span class="at">pattern =</span> <span class="st">&quot;*.csv&quot;</span>)) <span class="co"># names of colDatas </span></span>
<span id="cb15-2"><a href="data.html#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="data.html#cb15-3" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">substr</span>(CDs, <span class="dv">0</span>, <span class="fu">nchar</span>(CDs) <span class="sc">-</span> <span class="dv">4</span>)<span class="co"># assign name </span></span>
<span id="cb15-4"><a href="data.html#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="data.html#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># read in colData</span></span>
<span id="cb15-6"><a href="data.html#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(names)) {</span>
<span id="cb15-7"><a href="data.html#cb15-7" aria-hidden="true" tabindex="-1"></a>  name  <span class="ot">=</span> names[i]</span>
<span id="cb15-8"><a href="data.html#cb15-8" aria-hidden="true" tabindex="-1"></a>  value <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(<span class="st">&quot;./data/ColDatas/&quot;</span>, CDs[i]), <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb15-9"><a href="data.html#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dbWriteTable</span>(conn, name, value, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-10"><a href="data.html#cb15-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>Reading in differential expression files in <code>./data/DegData/</code> directory: You can store them into different sub-directories based on species, as illustrated here. A combined <code>deg_df</code> for each species is generated and stored, which makes meta-analysis and cross-species comparison easier.</p>
<p>In addition, since some experiments contain more than one differential expression analysis file, they are combined into one for each experiment, as illustrated below. To do this, you will need to store multiples into a folder for each study. A schematic illustration is shown below:</p>
<p><img src="data/Images/data_format.png" width="625" /></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="data.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># table names for files </span></span>
<span id="cb16-2"><a href="data.html#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="do">## assign names here</span></span>
<span id="cb16-3"><a href="data.html#cb16-3" aria-hidden="true" tabindex="-1"></a>deg_df_names <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;subtype_deg_df&quot;</span>, <span class="st">&quot;mouse_deg_df&quot;</span>, <span class="st">&quot;rat_deg_df&quot;</span>, <span class="st">&quot;ipsc_deg_df&quot;</span>, <span class="st">&quot;db_deg_df&quot;</span>, <span class="st">&quot;cts_deg_df&quot;</span>)</span>
<span id="cb16-4"><a href="data.html#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="data.html#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co"># you need to specify the names of the dataset (experiments) and store them in the list below</span></span>
<span id="cb16-6"><a href="data.html#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="do">## assign names here</span></span>
<span id="cb16-7"><a href="data.html#cb16-7" aria-hidden="true" tabindex="-1"></a>Dataset <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;mouse/Subtype DRG (Barry)/&quot;</span>, <span class="st">&quot;mouse/Mouse DRG Bulk/&quot;</span>, <span class="st">&quot;rat/Rat DRG Bulk/&quot;</span>,</span>
<span id="cb16-8"><a href="data.html#cb16-8" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;human/Human iPSC HSN1/&quot;</span>, <span class="st">&quot;human/Human skin Diabetes/&quot;</span>, <span class="st">&quot;human/Human skin Carpal Tunnel (CTS)/&quot;</span>)</span>
<span id="cb16-9"><a href="data.html#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="data.html#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="do">## just run this</span></span>
<span id="cb16-11"><a href="data.html#cb16-11" aria-hidden="true" tabindex="-1"></a>human_deg_df <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb16-12"><a href="data.html#cb16-12" aria-hidden="true" tabindex="-1"></a>mouse_deg_df <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb16-13"><a href="data.html#cb16-13" aria-hidden="true" tabindex="-1"></a>rat_deg_df   <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb16-14"><a href="data.html#cb16-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-15"><a href="data.html#cb16-15" aria-hidden="true" tabindex="-1"></a><span class="co"># for each experiment</span></span>
<span id="cb16-16"><a href="data.html#cb16-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(Dataset)) {</span>
<span id="cb16-17"><a href="data.html#cb16-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-18"><a href="data.html#cb16-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># name of experiment</span></span>
<span id="cb16-19"><a href="data.html#cb16-19" aria-hidden="true" tabindex="-1"></a>  name <span class="ot">=</span> Dataset[j] </span>
<span id="cb16-20"><a href="data.html#cb16-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-21"><a href="data.html#cb16-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># species  </span></span>
<span id="cb16-22"><a href="data.html#cb16-22" aria-hidden="true" tabindex="-1"></a>  species <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(name, <span class="at">split =</span> <span class="st">&quot;/&quot;</span>))[[<span class="dv">1</span>]]</span>
<span id="cb16-23"><a href="data.html#cb16-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-24"><a href="data.html#cb16-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># DEs are a list of datasets in this experiment </span></span>
<span id="cb16-25"><a href="data.html#cb16-25" aria-hidden="true" tabindex="-1"></a>  DEs <span class="ot">=</span> <span class="fu">str_sort</span>(<span class="fu">list.files</span>(<span class="at">path =</span> <span class="fu">paste</span>(<span class="st">&quot;./data/DegData/&quot;</span>, name, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), <span class="at">pattern =</span> <span class="st">&quot;*.csv&quot;</span>))</span>
<span id="cb16-26"><a href="data.html#cb16-26" aria-hidden="true" tabindex="-1"></a>  DE_names <span class="ot">&lt;-</span> <span class="fu">substr</span>(DEs, <span class="dv">0</span>, <span class="fu">nchar</span>(DEs) <span class="sc">-</span> <span class="dv">4</span>)<span class="co"># assign name to each population</span></span>
<span id="cb16-27"><a href="data.html#cb16-27" aria-hidden="true" tabindex="-1"></a>  deg_list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb16-28"><a href="data.html#cb16-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-29"><a href="data.html#cb16-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># read in those datasets into one dataframes for each experiment</span></span>
<span id="cb16-30"><a href="data.html#cb16-30" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(DE_names)) {</span>
<span id="cb16-31"><a href="data.html#cb16-31" aria-hidden="true" tabindex="-1"></a>    df_name <span class="ot">=</span> DE_names[i]</span>
<span id="cb16-32"><a href="data.html#cb16-32" aria-hidden="true" tabindex="-1"></a>    value   <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste</span>(<span class="st">&quot;./data/DegData/&quot;</span>, name, DEs[i], <span class="at">sep=</span><span class="st">&quot;&quot;</span>), <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb16-33"><a href="data.html#cb16-33" aria-hidden="true" tabindex="-1"></a>    deg_list <span class="ot">=</span> <span class="fu">append</span>(deg_list, <span class="fu">list</span>(value))</span>
<span id="cb16-34"><a href="data.html#cb16-34" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-35"><a href="data.html#cb16-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-36"><a href="data.html#cb16-36" aria-hidden="true" tabindex="-1"></a>  dataset_combine_deg <span class="ot">=</span> <span class="fu">generate_degdf</span>(deg_list, DE_names, <span class="fu">unlist</span>(<span class="fu">strsplit</span>(name, <span class="at">split =</span> <span class="st">&quot;/&quot;</span>))[[<span class="dv">2</span>]])</span>
<span id="cb16-37"><a href="data.html#cb16-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dbWriteTable</span>(conn, deg_df_names[j], dataset_combine_deg)</span>
<span id="cb16-38"><a href="data.html#cb16-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-39"><a href="data.html#cb16-39" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add to species-specific lists of data frames </span></span>
<span id="cb16-40"><a href="data.html#cb16-40" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (species <span class="sc">==</span> <span class="st">&quot;human&quot;</span>){</span>
<span id="cb16-41"><a href="data.html#cb16-41" aria-hidden="true" tabindex="-1"></a>    human_deg_df <span class="ot">=</span> <span class="fu">rbind</span>(human_deg_df, dataset_combine_deg)</span>
<span id="cb16-42"><a href="data.html#cb16-42" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-43"><a href="data.html#cb16-43" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (species <span class="sc">==</span> <span class="st">&quot;mouse&quot;</span>){</span>
<span id="cb16-44"><a href="data.html#cb16-44" aria-hidden="true" tabindex="-1"></a>    mouse_deg_df <span class="ot">=</span> <span class="fu">rbind</span>(mouse_deg_df, dataset_combine_deg)</span>
<span id="cb16-45"><a href="data.html#cb16-45" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-46"><a href="data.html#cb16-46" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (species <span class="sc">==</span> <span class="st">&quot;rat&quot;</span>){</span>
<span id="cb16-47"><a href="data.html#cb16-47" aria-hidden="true" tabindex="-1"></a>    rat_deg_df <span class="ot">=</span> <span class="fu">rbind</span>(rat_deg_df, dataset_combine_deg)</span>
<span id="cb16-48"><a href="data.html#cb16-48" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-49"><a href="data.html#cb16-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-50"><a href="data.html#cb16-50" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-51"><a href="data.html#cb16-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-52"><a href="data.html#cb16-52" aria-hidden="true" tabindex="-1"></a><span class="co"># write species deg_df into table</span></span>
<span id="cb16-53"><a href="data.html#cb16-53" aria-hidden="true" tabindex="-1"></a><span class="fu">dbWriteTable</span>(conn, <span class="st">&quot;mouse_all_deg_df&quot;</span>, mouse_deg_df)</span>
<span id="cb16-54"><a href="data.html#cb16-54" aria-hidden="true" tabindex="-1"></a><span class="fu">dbWriteTable</span>(conn, <span class="st">&quot;human_deg_df&quot;</span>, human_deg_df)</span>
<span id="cb16-55"><a href="data.html#cb16-55" aria-hidden="true" tabindex="-1"></a><span class="fu">dbWriteTable</span>(conn, <span class="st">&quot;rat_deg_df&quot;</span>, rat_deg_df)</span></code></pre></div>
<p>Now, the <code>.db</code> object should contain all data required to run your app.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="ui.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
