<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Server | Sequencing databases in Shiny</title>
  <meta name="description" content="Here we will describe how to build a Shiny app for sequencing data using the code base described in CITE PAPER HERE." />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Server | Sequencing databases in Shiny" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Here we will describe how to build a Shiny app for sequencing data using the code base described in CITE PAPER HERE." />
  <meta name="github-repo" content="aliibarry/omics-database" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Server | Sequencing databases in Shiny" />
  
  <meta name="twitter:description" content="Here we will describe how to build a Shiny app for sequencing data using the code base described in CITE PAPER HERE." />
  

<meta name="author" content="AMB and NZ" />


<meta name="date" content="2023-03-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="ui.html"/>
<link rel="next" href="deploy.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Sequencing databases in Shiny</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#shiny"><i class="fa fa-check"></i><b>1.1</b> Shiny</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow"><i class="fa fa-check"></i><b>1.2</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>2</b> Data Processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data.html"><a href="data.html#preparation"><i class="fa fa-check"></i><b>2.1</b> Preparation</a></li>
<li class="chapter" data-level="2.2" data-path="data.html"><a href="data.html#species-optional"><i class="fa fa-check"></i><b>2.2</b> Species <em>(optional)</em></a></li>
<li class="chapter" data-level="2.3" data-path="data.html"><a href="data.html#rdata-option-1"><i class="fa fa-check"></i><b>2.3</b> RData <em>(option 1)</em></a></li>
<li class="chapter" data-level="2.4" data-path="data.html"><a href="data.html#building-an-sql-database-via-r-recommended"><i class="fa fa-check"></i><b>2.4</b> Building an SQL database via R <em>(recommended)</em></a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="data.html"><a href="data.html#sql-manipulation"><i class="fa fa-check"></i><b>2.4.1</b> SQL manipulation</a></li>
<li class="chapter" data-level="2.4.2" data-path="data.html"><a href="data.html#processing-multiple-files"><i class="fa fa-check"></i><b>2.4.2</b> Processing multiple files</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="ui.html"><a href="ui.html"><i class="fa fa-check"></i><b>3</b> User Interface</a>
<ul>
<li class="chapter" data-level="3.1" data-path="ui.html"><a href="ui.html#basic-structure"><i class="fa fa-check"></i><b>3.1</b> Basic structure</a></li>
<li class="chapter" data-level="3.2" data-path="ui.html"><a href="ui.html#header"><i class="fa fa-check"></i><b>3.2</b> Header</a></li>
<li class="chapter" data-level="3.3" data-path="ui.html"><a href="ui.html#sidebar"><i class="fa fa-check"></i><b>3.3</b> Sidebar</a></li>
<li class="chapter" data-level="3.4" data-path="ui.html"><a href="ui.html#body"><i class="fa fa-check"></i><b>3.4</b> Body</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="ui.html"><a href="ui.html#homepage"><i class="fa fa-check"></i><b>3.4.1</b> Homepage</a></li>
<li class="chapter" data-level="3.4.2" data-path="ui.html"><a href="ui.html#grouped-comparisons"><i class="fa fa-check"></i><b>3.4.2</b> Grouped comparisons</a></li>
<li class="chapter" data-level="3.4.3" data-path="ui.html"><a href="ui.html#individual-analysis"><i class="fa fa-check"></i><b>3.4.3</b> Individual Analysis</a></li>
<li class="chapter" data-level="3.4.4" data-path="ui.html"><a href="ui.html#network-page"><i class="fa fa-check"></i><b>3.4.4</b> Network page</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="server.html"><a href="server.html"><i class="fa fa-check"></i><b>4</b> Server</a>
<ul>
<li class="chapter" data-level="4.1" data-path="server.html"><a href="server.html#requirements"><i class="fa fa-check"></i><b>4.1</b> Requirements</a></li>
<li class="chapter" data-level="4.2" data-path="server.html"><a href="server.html#server-overview"><i class="fa fa-check"></i><b>4.2</b> Server Overview</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="server.html"><a href="server.html#side-bar"><i class="fa fa-check"></i><b>4.2.1</b> Side bar</a></li>
<li class="chapter" data-level="4.2.2" data-path="server.html"><a href="server.html#tab-menu"><i class="fa fa-check"></i><b>4.2.2</b> Tab Menu</a></li>
<li class="chapter" data-level="4.2.3" data-path="server.html"><a href="server.html#grouped-comparisons-1"><i class="fa fa-check"></i><b>4.2.3</b> Grouped comparisons</a></li>
<li class="chapter" data-level="4.2.4" data-path="server.html"><a href="server.html#individual-analysis-1"><i class="fa fa-check"></i><b>4.2.4</b> Individual Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="server.html"><a href="server.html#network-analysis"><i class="fa fa-check"></i><b>4.3</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="server.html"><a href="server.html#integration-with-string-db"><i class="fa fa-check"></i><b>4.3.1</b> Integration with STRING DB</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="server.html"><a href="server.html#functions"><i class="fa fa-check"></i><b>4.4</b> Functions</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="server.html"><a href="server.html#data-processing"><i class="fa fa-check"></i><b>4.4.1</b> Data processing</a></li>
<li class="chapter" data-level="4.4.2" data-path="server.html"><a href="server.html#dot-plots"><i class="fa fa-check"></i><b>4.4.2</b> Dot plots</a></li>
<li class="chapter" data-level="4.4.3" data-path="server.html"><a href="server.html#line-plots"><i class="fa fa-check"></i><b>4.4.3</b> Line plots</a></li>
<li class="chapter" data-level="4.4.4" data-path="server.html"><a href="server.html#a-differential-expression-dot-plot"><i class="fa fa-check"></i><b>4.4.4</b> A differential expression dot plot</a></li>
<li class="chapter" data-level="4.4.5" data-path="server.html"><a href="server.html#a-volcano-plot"><i class="fa fa-check"></i><b>4.4.5</b> A volcano plot</a></li>
<li class="chapter" data-level="4.4.6" data-path="server.html"><a href="server.html#other"><i class="fa fa-check"></i><b>4.4.6</b> Other</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="deploy.html"><a href="deploy.html"><i class="fa fa-check"></i><b>5</b> Deploying your application</a>
<ul>
<li class="chapter" data-level="5.1" data-path="deploy.html"><a href="deploy.html#shinyapps.io"><i class="fa fa-check"></i><b>5.1</b> Shinyapps.io</a></li>
<li class="chapter" data-level="5.2" data-path="deploy.html"><a href="deploy.html#bioconductor"><i class="fa fa-check"></i><b>5.2</b> Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="trouble.html"><a href="trouble.html"><i class="fa fa-check"></i><b>6</b> Troubleshooting</a>
<ul>
<li class="chapter" data-level="6.1" data-path="trouble.html"><a href="trouble.html#package-installation"><i class="fa fa-check"></i><b>6.1</b> Package installation</a></li>
<li class="chapter" data-level="6.2" data-path="trouble.html"><a href="trouble.html#file-setup"><i class="fa fa-check"></i><b>6.2</b> File setup</a></li>
<li class="chapter" data-level="6.3" data-path="trouble.html"><a href="trouble.html#styling"><i class="fa fa-check"></i><b>6.3</b> Styling</a></li>
<li class="chapter" data-level="6.4" data-path="trouble.html"><a href="trouble.html#data-inputoutput-issues"><i class="fa fa-check"></i><b>6.4</b> Data input/output issues</a></li>
<li class="chapter" data-level="6.5" data-path="trouble.html"><a href="trouble.html#debugging-reactive-code"><i class="fa fa-check"></i><b>6.5</b> Debugging reactive code</a></li>
<li class="chapter" data-level="6.6" data-path="trouble.html"><a href="trouble.html#deployment"><i class="fa fa-check"></i><b>6.6</b> Deployment</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="trouble.html"><a href="trouble.html#basics"><i class="fa fa-check"></i><b>6.6.1</b> Basics</a></li>
<li class="chapter" data-level="6.6.2" data-path="trouble.html"><a href="trouble.html#bioconductor-1"><i class="fa fa-check"></i><b>6.6.2</b> Bioconductor</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Sequencing databases in Shiny</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="server" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Server<a href="server.html#server" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<style>
img
{
    display:block;
    float:none;
    margin-left:auto;
    margin-right:auto;
}
</style>
<p>Once you:<br />
1. Have created a dataset description <code>des_df</code> object<br />
2. Have changed the directory of the <code>.db</code> to the one you have stored your data in</p>
<p>…you can start running the app.</p>
<p><em>(but lots of details are provided below so you can dive into the code)</em></p>
<div id="requirements" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Requirements<a href="server.html#requirements" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For the <code>Server.R</code> page to run, you will need to create a <code>des_df</code>, a description df that describes your datasets. An example is shown below:</p>
<ul>
<li><strong>Dataset</strong>: The name of the dataset</li>
<li><strong>disease_condition</strong>: The experimental disease group name</li>
<li><strong>species</strong>: the species in the experiment</li>
<li><strong>count_df_name</strong>: the name of the count matrix</li>
<li><strong>colData_name</strong>: The name of experimental design</li>
<li><strong>deg_df_name</strong>: the name of differential expression data</li>
<li><strong>type</strong>: Name of the RNA sequencing technique; either bulk or single</li>
<li><strong>identifier</strong>: This identifier is useful for loading description files specific for each experiment. E.g. subtype.Rhtml . Name your description files and add their names to this column.</li>
<li><strong>include_subtypes</strong>: This specifies if there are different subtypes of samples in the experiment. E.g. different species strains. If TRUE, there will be an additional line plot showing subtype-specific changes in gene expression.</li>
<li><strong>include_degs</strong>: This specifies if there are differential expression data files for this experiment. If TRUE, the visualisation page will yield differential expression dot plots and volcano plots.</li>
<li><strong>include_gois</strong>: Whether to display the count table on the site; i.e. some studies utilises human patient data which are sensitive to disclose. For studies like this, include_gois will be FALSE.</li>
</ul>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="server.html#cb26-1" aria-hidden="true" tabindex="-1"></a>des_df <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb26-2"><a href="server.html#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">Dataset =</span> <span class="fu">c</span>(<span class="st">&quot;Subtype DRG (Barry)&quot;</span>, <span class="st">&quot;Mouse DRG Bulk&quot;</span>, <span class="st">&quot;Rat DRG Bulk&quot;</span>,</span>
<span id="cb26-3"><a href="server.html#cb26-3" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;Human iPSC HSN1&quot;</span>, <span class="st">&quot;Human skin Diabetes&quot;</span>, <span class="st">&quot;Human skin Carpal Tunnel (CTS)&quot;</span>,</span>
<span id="cb26-4"><a href="server.html#cb26-4" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;Mouse DRG Subtype (Zheng)&quot;</span>,<span class="st">&quot;Human DRG RNAseq&quot;</span>, <span class="st">&quot;Human DRG spatial-seq&quot;</span>),</span>
<span id="cb26-5"><a href="server.html#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">disease_condition =</span> <span class="fu">c</span>(<span class="st">&quot;ipsi&quot;</span>, <span class="st">&quot;SNI&quot;</span>, <span class="st">&quot;SNT&quot;</span>, <span class="st">&quot;patient&quot;</span>, <span class="st">&quot;PDPN&quot;</span>, <span class="st">&quot;pre_Surgery&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;P&quot;</span>, <span class="st">&quot;&quot;</span>),</span>
<span id="cb26-6"><a href="server.html#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="fu">c</span>(<span class="st">&quot;mouse&quot;</span>, <span class="st">&quot;mouse&quot;</span>, <span class="st">&quot;rat&quot;</span>, <span class="st">&quot;human&quot;</span>, <span class="st">&quot;human&quot;</span>, <span class="st">&quot;human&quot;</span>, <span class="st">&quot;mouse&quot;</span>, <span class="st">&quot;human&quot;</span>, <span class="st">&quot;human&quot;</span>),</span>
<span id="cb26-7"><a href="server.html#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">count_df_name =</span> <span class="fu">c</span>(<span class="st">&quot;TPM_subtype&quot;</span>, <span class="st">&quot;TPM_mouse&quot;</span>, <span class="st">&quot;TPM_rat&quot;</span>, <span class="st">&quot;TPM_ipsc&quot;</span>, <span class="st">&quot;TPM_HS_diabetes&quot;</span>, <span class="st">&quot;TPM_HS_CTS&quot;</span>, <span class="st">&quot;TPM_zheng&quot;</span>, <span class="st">&quot;TPM_humandrg&quot;</span>, <span class="st">&quot;&quot;</span>),</span>
<span id="cb26-8"><a href="server.html#cb26-8" aria-hidden="true" tabindex="-1"></a>   <span class="at">colData_name =</span> <span class="fu">c</span>(<span class="st">&quot;bulkseq_colData&quot;</span>, <span class="st">&quot;TPM_mouse_colData&quot;</span>, <span class="st">&quot;TPM_rat_colData&quot;</span>, <span class="st">&quot;ipsc_colData&quot;</span>, <span class="st">&quot;db_colData&quot;</span>, <span class="st">&quot;skin_colData&quot;</span>, <span class="st">&quot;zheng_colData&quot;</span>, <span class="st">&quot;humandrg_colData&quot;</span>, <span class="st">&quot;&quot;</span>),</span>
<span id="cb26-9"><a href="server.html#cb26-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">deg_df_name =</span> <span class="fu">c</span>(<span class="st">&quot;subtype_deg_df&quot;</span>, <span class="st">&quot;mouse_deg_df&quot;</span>, <span class="st">&quot;rat_deg_df&quot;</span>, <span class="st">&quot;ipsc_deg_df&quot;</span>, <span class="st">&quot;db_deg_df&quot;</span>, <span class="st">&quot;cts_deg_df&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;&quot;</span>),</span>
<span id="cb26-10"><a href="server.html#cb26-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="fu">c</span>(<span class="st">&quot;bulk&quot;</span>,<span class="st">&quot;bulk&quot;</span>,<span class="st">&quot;bulk&quot;</span>,<span class="st">&quot;bulk&quot;</span>,<span class="st">&quot;bulk&quot;</span>,<span class="st">&quot;bulk&quot;</span>,<span class="st">&quot;bulk&quot;</span>,<span class="st">&quot;bulk&quot;</span>, <span class="st">&quot;single&quot;</span>),</span>
<span id="cb26-11"><a href="server.html#cb26-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">identifier =</span> <span class="fu">c</span>(<span class="st">&quot;subtype&quot;</span>, <span class="st">&quot;mouse&quot;</span>, <span class="st">&quot;rat&quot;</span>, <span class="st">&quot;ipsc&quot;</span>, <span class="st">&quot;HS_diabetes&quot;</span>, <span class="st">&quot;HS_CTS&quot;</span>, <span class="st">&quot;zheng&quot;</span>, <span class="st">&quot;humandrg&quot;</span>, <span class="st">&quot;scrna&quot;</span>),</span>
<span id="cb26-12"><a href="server.html#cb26-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">include_subtypes =</span> <span class="fu">c</span>(<span class="cn">TRUE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span>),</span>
<span id="cb26-13"><a href="server.html#cb26-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">include_degs =</span> <span class="fu">c</span>(<span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span>),</span>
<span id="cb26-14"><a href="server.html#cb26-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">include_gois =</span> <span class="fu">c</span>(<span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="cn">TRUE</span>))</span>
<span id="cb26-15"><a href="server.html#cb26-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-16"><a href="server.html#cb26-16" aria-hidden="true" tabindex="-1"></a><span class="fu">kbl</span>(des_df <span class="sc">%&gt;%</span> head) <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;condensed&quot;</span>, <span class="st">&quot;responsive&quot;</span>), <span class="at">font_size =</span> <span class="dv">12</span>, <span class="at">latex_options =</span> <span class="st">&quot;scale_down&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;100%&quot;</span>, <span class="at">height =</span> <span class="st">&quot;500px&quot;</span>)</span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:100%; ">
<table class="table table-striped table-hover table-condensed table-responsive" style="font-size: 12px; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Dataset
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
disease_condition
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
species
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
count_df_name
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
colData_name
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
deg_df_name
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
type
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
identifier
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
include_subtypes
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
include_degs
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
include_gois
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Subtype DRG (Barry)
</td>
<td style="text-align:left;">
ipsi
</td>
<td style="text-align:left;">
mouse
</td>
<td style="text-align:left;">
TPM_subtype
</td>
<td style="text-align:left;">
bulkseq_colData
</td>
<td style="text-align:left;">
subtype_deg_df
</td>
<td style="text-align:left;">
bulk
</td>
<td style="text-align:left;">
subtype
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
Mouse DRG Bulk
</td>
<td style="text-align:left;">
SNI
</td>
<td style="text-align:left;">
mouse
</td>
<td style="text-align:left;">
TPM_mouse
</td>
<td style="text-align:left;">
TPM_mouse_colData
</td>
<td style="text-align:left;">
mouse_deg_df
</td>
<td style="text-align:left;">
bulk
</td>
<td style="text-align:left;">
mouse
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
Rat DRG Bulk
</td>
<td style="text-align:left;">
SNT
</td>
<td style="text-align:left;">
rat
</td>
<td style="text-align:left;">
TPM_rat
</td>
<td style="text-align:left;">
TPM_rat_colData
</td>
<td style="text-align:left;">
rat_deg_df
</td>
<td style="text-align:left;">
bulk
</td>
<td style="text-align:left;">
rat
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
Human iPSC HSN1
</td>
<td style="text-align:left;">
patient
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
TPM_ipsc
</td>
<td style="text-align:left;">
ipsc_colData
</td>
<td style="text-align:left;">
ipsc_deg_df
</td>
<td style="text-align:left;">
bulk
</td>
<td style="text-align:left;">
ipsc
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
Human skin Diabetes
</td>
<td style="text-align:left;">
PDPN
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
TPM_HS_diabetes
</td>
<td style="text-align:left;">
db_colData
</td>
<td style="text-align:left;">
db_deg_df
</td>
<td style="text-align:left;">
bulk
</td>
<td style="text-align:left;">
HS_diabetes
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
Human skin Carpal Tunnel (CTS)
</td>
<td style="text-align:left;">
pre_Surgery
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
TPM_HS_CTS
</td>
<td style="text-align:left;">
skin_colData
</td>
<td style="text-align:left;">
cts_deg_df
</td>
<td style="text-align:left;">
bulk
</td>
<td style="text-align:left;">
HS_CTS
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
</tbody>
</table>
</div>
<p> </p>
<p>The table names of <code>countdata</code> and <code>colData</code> for each experiment will be written. This allows the app to directly query those tables from the database object. In addition, you will need to specify directory that storing your data</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="server.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="do">## directory for scRNA sequencing data; leave it for &quot;&quot; if none </span></span>
<span id="cb27-2"><a href="server.html#cb27-2" aria-hidden="true" tabindex="-1"></a>scRNA_dir <span class="ot">=</span> <span class="st">&quot;data/drg.combined.rds&quot;</span></span>
<span id="cb27-3"><a href="server.html#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="server.html#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="do">## a file containing themes for plots and labels for axis </span></span>
<span id="cb27-5"><a href="server.html#cb27-5" aria-hidden="true" tabindex="-1"></a>theme_dir <span class="ot">=</span> <span class="st">&quot;global.R&quot;</span></span>
<span id="cb27-6"><a href="server.html#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="server.html#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="do">## RData that contains DEG files for network analysis </span></span>
<span id="cb27-8"><a href="server.html#cb27-8" aria-hidden="true" tabindex="-1"></a>network_dir <span class="ot">=</span> <span class="st">&quot;data/network.RData&quot;</span></span>
<span id="cb27-9"><a href="server.html#cb27-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-10"><a href="server.html#cb27-10" aria-hidden="true" tabindex="-1"></a><span class="do">## directory for the sql database </span></span>
<span id="cb27-11"><a href="server.html#cb27-11" aria-hidden="true" tabindex="-1"></a>db_dir <span class="ot">=</span> <span class="st">&quot;test.db&quot;</span></span></code></pre></div>
</div>
<div id="server-overview" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Server Overview<a href="server.html#server-overview" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><em>You don’t need to modify this, but is provided as an overview of the app structure</em>.</p>
<p><code>server()</code> function is invoked each time a new session starts. When the server function is called it creates a new local environment that is independent of every other invocation of the function. This allows each session to have a unique state, as well as isolating the variables created inside the function.</p>
<p>Server functions take three parameters: <code>input</code>, <code>output</code>, and <code>session</code>. The input argument is a list-like object that contains all the input data sent from <code>ui.R</code>, named according to the input ID.</p>
<div id="side-bar" class="section level3 hasAnchor" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Side bar<a href="server.html#side-bar" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Update the gene symbols entered or uploaded in files into reactive variables.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="server.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># gene search bar</span></span>
<span id="cb28-2"><a href="server.html#cb28-2" aria-hidden="true" tabindex="-1"></a>shiny<span class="sc">::</span><span class="fu">updateSelectizeInput</span>(session,</span>
<span id="cb28-3"><a href="server.html#cb28-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">inputId =</span> <span class="st">&quot;geneid&quot;</span>,</span>
<span id="cb28-4"><a href="server.html#cb28-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">label =</span> <span class="st">&quot;Search Genes:&quot;</span>,</span>
<span id="cb28-5"><a href="server.html#cb28-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">choices =</span> allgenes,</span>
<span id="cb28-6"><a href="server.html#cb28-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">server =</span> <span class="cn">TRUE</span>,</span>
<span id="cb28-7"><a href="server.html#cb28-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">selected =</span> <span class="fu">c</span>(<span class="st">&quot;Trpv1&quot;</span>, <span class="st">&quot;Scn10a&quot;</span>,<span class="st">&quot;Atf3&quot;</span>) <span class="do">## can change to relevant genes</span></span>
<span id="cb28-8"><a href="server.html#cb28-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb28-9"><a href="server.html#cb28-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-10"><a href="server.html#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="co"># a reactive variable that records whether a file is uploaded.</span></span>
<span id="cb28-11"><a href="server.html#cb28-11" aria-hidden="true" tabindex="-1"></a>rv <span class="ot">&lt;-</span> <span class="fu">reactiveValues</span>(</span>
<span id="cb28-12"><a href="server.html#cb28-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">clear =</span> <span class="cn">FALSE</span>,</span>
<span id="cb28-13"><a href="server.html#cb28-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> <span class="cn">FALSE</span></span>
<span id="cb28-14"><a href="server.html#cb28-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb28-15"><a href="server.html#cb28-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-16"><a href="server.html#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="co"># read in files uploaded </span></span>
<span id="cb28-17"><a href="server.html#cb28-17" aria-hidden="true" tabindex="-1"></a>shiny<span class="sc">::</span><span class="fu">observeEvent</span>(input<span class="sc">$</span>file, {</span>
<span id="cb28-18"><a href="server.html#cb28-18" aria-hidden="true" tabindex="-1"></a>  rv<span class="sc">$</span>clear <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb28-19"><a href="server.html#cb28-19" aria-hidden="true" tabindex="-1"></a>  rv<span class="sc">$</span>data <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb28-20"><a href="server.html#cb28-20" aria-hidden="true" tabindex="-1"></a>}, <span class="at">priority =</span> <span class="dv">1000</span>)</span>
<span id="cb28-21"><a href="server.html#cb28-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-22"><a href="server.html#cb28-22" aria-hidden="true" tabindex="-1"></a><span class="co"># clear the file space when files are removed</span></span>
<span id="cb28-23"><a href="server.html#cb28-23" aria-hidden="true" tabindex="-1"></a>shiny<span class="sc">::</span><span class="fu">observeEvent</span>(input<span class="sc">$</span>reset, {</span>
<span id="cb28-24"><a href="server.html#cb28-24" aria-hidden="true" tabindex="-1"></a>  rv<span class="sc">$</span>clear <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb28-25"><a href="server.html#cb28-25" aria-hidden="true" tabindex="-1"></a>  rv<span class="sc">$</span>data <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb28-26"><a href="server.html#cb28-26" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb28-27"><a href="server.html#cb28-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-28"><a href="server.html#cb28-28" aria-hidden="true" tabindex="-1"></a><span class="co"># read into gene ids from files</span></span>
<span id="cb28-29"><a href="server.html#cb28-29" aria-hidden="true" tabindex="-1"></a>file_input <span class="ot">&lt;-</span> <span class="fu">reactive</span>({</span>
<span id="cb28-30"><a href="server.html#cb28-30" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (rv<span class="sc">$</span>clear <span class="sc">==</span> <span class="cn">TRUE</span>) {</span>
<span id="cb28-31"><a href="server.html#cb28-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb28-32"><a href="server.html#cb28-32" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb28-33"><a href="server.html#cb28-33" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(rv<span class="sc">$</span>clear<span class="sc">==</span><span class="cn">FALSE</span> <span class="sc">&amp;&amp;</span> rv<span class="sc">$</span>data <span class="sc">==</span> <span class="cn">TRUE</span>) {</span>
<span id="cb28-34"><a href="server.html#cb28-34" aria-hidden="true" tabindex="-1"></a>    goi <span class="ot">=</span> <span class="fu">read.table</span>(input<span class="sc">$</span>file<span class="sc">$</span>datapath)</span>
<span id="cb28-35"><a href="server.html#cb28-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(goi) <span class="ot">&lt;-</span> goi[,<span class="dv">1</span>]</span>
<span id="cb28-36"><a href="server.html#cb28-36" aria-hidden="true" tabindex="-1"></a>    goi <span class="ot">&lt;-</span> goi[<span class="fu">which</span>(<span class="fu">rownames</span>(goi) <span class="sc">%in%</span> allgenes<span class="sc">==</span><span class="cn">TRUE</span>),]</span>
<span id="cb28-37"><a href="server.html#cb28-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(goi)}</span>
<span id="cb28-38"><a href="server.html#cb28-38" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb28-39"><a href="server.html#cb28-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-40"><a href="server.html#cb28-40" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">=</span> <span class="fu">reactive</span>({</span>
<span id="cb28-41"><a href="server.html#cb28-41" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(<span class="fu">file_input</span>())) {</span>
<span id="cb28-42"><a href="server.html#cb28-42" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> input<span class="sc">$</span>geneid</span>
<span id="cb28-43"><a href="server.html#cb28-43" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb28-44"><a href="server.html#cb28-44" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> {</span>
<span id="cb28-45"><a href="server.html#cb28-45" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> <span class="fu">file_input</span>()</span>
<span id="cb28-46"><a href="server.html#cb28-46" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb28-47"><a href="server.html#cb28-47" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
</div>
<div id="tab-menu" class="section level3 hasAnchor" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Tab Menu<a href="server.html#tab-menu" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Update active tab based on the current selection of a tab menu.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="server.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="do">### this doesn&#39;t need to be modified</span></span>
<span id="cb29-2"><a href="server.html#cb29-2" aria-hidden="true" tabindex="-1"></a>shiny<span class="sc">::</span><span class="fu">observeEvent</span>(input<span class="sc">$</span>metaa, {</span>
<span id="cb29-3"><a href="server.html#cb29-3" aria-hidden="true" tabindex="-1"></a>  Newtab <span class="ot">&lt;-</span> <span class="cf">switch</span>(input<span class="sc">$</span>tabs, <span class="st">&quot;tabhome&quot;</span> <span class="ot">=</span> <span class="st">&quot;tabmeta&quot;</span>)</span>
<span id="cb29-4"><a href="server.html#cb29-4" aria-hidden="true" tabindex="-1"></a>  shinydashboard<span class="sc">::</span><span class="fu">updateTabItems</span>(session, <span class="st">&quot;tabs&quot;</span>, Newtab) </span>
<span id="cb29-5"><a href="server.html#cb29-5" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb29-6"><a href="server.html#cb29-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-7"><a href="server.html#cb29-7" aria-hidden="true" tabindex="-1"></a>shiny<span class="sc">::</span><span class="fu">observeEvent</span>(input<span class="sc">$</span>inda, {</span>
<span id="cb29-8"><a href="server.html#cb29-8" aria-hidden="true" tabindex="-1"></a>  Newtab <span class="ot">&lt;-</span> <span class="cf">switch</span>(input<span class="sc">$</span>tabs,<span class="st">&quot;tabhome&quot;</span> <span class="ot">=</span> <span class="st">&quot;tabdata&quot;</span>)</span>
<span id="cb29-9"><a href="server.html#cb29-9" aria-hidden="true" tabindex="-1"></a>  shinydashboard<span class="sc">::</span><span class="fu">updateTabItems</span>(session, <span class="st">&quot;tabs&quot;</span>, Newtab) </span>
<span id="cb29-10"><a href="server.html#cb29-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb29-11"><a href="server.html#cb29-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-12"><a href="server.html#cb29-12" aria-hidden="true" tabindex="-1"></a>shiny<span class="sc">::</span><span class="fu">observeEvent</span>(input<span class="sc">$</span>neta, {</span>
<span id="cb29-13"><a href="server.html#cb29-13" aria-hidden="true" tabindex="-1"></a>  Newtab <span class="ot">&lt;-</span> <span class="cf">switch</span>(input<span class="sc">$</span>tabs,<span class="st">&quot;tabhome&quot;</span> <span class="ot">=</span> <span class="st">&quot;tabnet&quot;</span>)</span>
<span id="cb29-14"><a href="server.html#cb29-14" aria-hidden="true" tabindex="-1"></a>  shinydashboard<span class="sc">::</span><span class="fu">updateTabItems</span>(session, <span class="st">&quot;tabs&quot;</span>, Newtab) </span>
<span id="cb29-15"><a href="server.html#cb29-15" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
</div>
<div id="grouped-comparisons-1" class="section level3 hasAnchor" number="4.2.3">
<h3><span class="header-section-number">4.2.3</span> Grouped comparisons<a href="server.html#grouped-comparisons-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>By using the same visualization tools and methods across studies, it becomes easier to compare and integrate results from different studies.</p>
<p><img src="data/Images/meta.png" width="3614" /></p>
<ol style="list-style-type: decimal">
<li>Render a table for datasets included</li>
</ol>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="server.html#cb30-1" aria-hidden="true" tabindex="-1"></a>output<span class="sc">$</span>meta_table <span class="ot">=</span> DT<span class="sc">::</span><span class="fu">renderDataTable</span>({</span>
<span id="cb30-2"><a href="server.html#cb30-2" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>(</span>
<span id="cb30-3"><a href="server.html#cb30-3" aria-hidden="true" tabindex="-1"></a>      bulk_df[<span class="fu">c</span>(<span class="st">&quot;Dataset&quot;</span>, <span class="st">&quot;Pain_Model&quot;</span>, <span class="st">&quot;Species&quot;</span>, <span class="st">&quot;Tissue&quot;</span>)], <span class="do">## can select relevant columns</span></span>
<span id="cb30-4"><a href="server.html#cb30-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">width =</span> <span class="dv">6</span>,</span>
<span id="cb30-5"><a href="server.html#cb30-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">class =</span> <span class="st">&#39;nowrap&#39;</span>,</span>
<span id="cb30-6"><a href="server.html#cb30-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX =</span> <span class="cn">TRUE</span>, <span class="at">scrollY =</span> <span class="cn">TRUE</span>, <span class="at">pageLength =</span> <span class="dv">5</span>), </span>
<span id="cb30-7"><a href="server.html#cb30-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">selection =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb30-8"><a href="server.html#cb30-8" aria-hidden="true" tabindex="-1"></a>      <span class="co"># selection = list(mode = &quot;multiple&quot;, selected = list(rows = c(1:nrow(bulk_df)), cols = c(1:4)), target = &quot;row+column&quot;)</span></span>
<span id="cb30-9"><a href="server.html#cb30-9" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb30-10"><a href="server.html#cb30-10" aria-hidden="true" tabindex="-1"></a>  })</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Generate homepage plots</li>
</ol>
<ul>
<li>A combined gene expression dot plots</li>
<li>A scRNA dotplot based on a <code>Seurat</code> object</li>
<li>A differential expression dot plot that allows visualisation of the significance level of differential gene expression</li>
</ul>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="server.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="do">### this doesn&#39;t need to be modified</span></span>
<span id="cb31-2"><a href="server.html#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="co"># plot in reponse to the &#39;load&#39; button </span></span>
<span id="cb31-3"><a href="server.html#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="fu">observeEvent</span>(input<span class="sc">$</span>load, {</span>
<span id="cb31-4"><a href="server.html#cb31-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-5"><a href="server.html#cb31-5" aria-hidden="true" tabindex="-1"></a>  file_input <span class="ot">&lt;-</span> <span class="fu">reactive</span>({</span>
<span id="cb31-6"><a href="server.html#cb31-6" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (rv<span class="sc">$</span>clear <span class="sc">==</span> <span class="cn">TRUE</span>) {</span>
<span id="cb31-7"><a href="server.html#cb31-7" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb31-8"><a href="server.html#cb31-8" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb31-9"><a href="server.html#cb31-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(rv<span class="sc">$</span>clear<span class="sc">==</span><span class="cn">FALSE</span> <span class="sc">&amp;&amp;</span> rv<span class="sc">$</span>data <span class="sc">==</span> <span class="cn">TRUE</span>) {</span>
<span id="cb31-10"><a href="server.html#cb31-10" aria-hidden="true" tabindex="-1"></a>      goi <span class="ot">=</span> <span class="fu">read.table</span>(input<span class="sc">$</span>file<span class="sc">$</span>datapath)</span>
<span id="cb31-11"><a href="server.html#cb31-11" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rownames</span>(goi) <span class="ot">&lt;-</span> goi[,<span class="dv">1</span>]</span>
<span id="cb31-12"><a href="server.html#cb31-12" aria-hidden="true" tabindex="-1"></a>      goi <span class="ot">&lt;-</span> goi[<span class="fu">which</span>(<span class="fu">rownames</span>(goi) <span class="sc">%in%</span> allgenes<span class="sc">==</span><span class="cn">TRUE</span>),]</span>
<span id="cb31-13"><a href="server.html#cb31-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(goi)}</span>
<span id="cb31-14"><a href="server.html#cb31-14" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb31-15"><a href="server.html#cb31-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-16"><a href="server.html#cb31-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(<span class="fu">file_input</span>())) {</span>
<span id="cb31-17"><a href="server.html#cb31-17" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> input<span class="sc">$</span>geneid</span>
<span id="cb31-18"><a href="server.html#cb31-18" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb31-19"><a href="server.html#cb31-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> {</span>
<span id="cb31-20"><a href="server.html#cb31-20" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> <span class="fu">file_input</span>()</span>
<span id="cb31-21"><a href="server.html#cb31-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb31-22"><a href="server.html#cb31-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-23"><a href="server.html#cb31-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># retrieve count data of selected datasets </span></span>
<span id="cb31-24"><a href="server.html#cb31-24" aria-hidden="true" tabindex="-1"></a>  df_list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb31-25"><a href="server.html#cb31-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-26"><a href="server.html#cb31-26" aria-hidden="true" tabindex="-1"></a>  selected_data <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(bulk_df))</span>
<span id="cb31-27"><a href="server.html#cb31-27" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> selected_data){</span>
<span id="cb31-28"><a href="server.html#cb31-28" aria-hidden="true" tabindex="-1"></a>    curr <span class="ot">=</span> bulk_df[i,]</span>
<span id="cb31-29"><a href="server.html#cb31-29" aria-hidden="true" tabindex="-1"></a>    d    <span class="ot">=</span> curr<span class="sc">$</span>count</span>
<span id="cb31-30"><a href="server.html#cb31-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (d<span class="sc">!=</span><span class="st">&quot;&quot;</span>){</span>
<span id="cb31-31"><a href="server.html#cb31-31" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (curr<span class="sc">$</span>Species <span class="sc">==</span> <span class="st">&quot;human&quot;</span>){</span>
<span id="cb31-32"><a href="server.html#cb31-32" aria-hidden="true" tabindex="-1"></a>        sql <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;SELECT * FROM&quot;</span>, d, <span class="st">&quot;WHERE mgi_symbol = ?&quot;</span>)</span>
<span id="cb31-33"><a href="server.html#cb31-33" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb31-34"><a href="server.html#cb31-34" aria-hidden="true" tabindex="-1"></a>      <span class="cf">else</span> {</span>
<span id="cb31-35"><a href="server.html#cb31-35" aria-hidden="true" tabindex="-1"></a>        sql <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;SELECT * FROM&quot;</span>, d, <span class="st">&quot;WHERE symbol = ?&quot;</span>)</span>
<span id="cb31-36"><a href="server.html#cb31-36" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb31-37"><a href="server.html#cb31-37" aria-hidden="true" tabindex="-1"></a>      count_data <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">dbGetPreparedQuery</span>(conn, sql, <span class="at">bind.data=</span><span class="fu">data.frame</span>(<span class="at">symbol=</span>genes)))</span>
<span id="cb31-38"><a href="server.html#cb31-38" aria-hidden="true" tabindex="-1"></a>      df_list    <span class="ot">=</span> <span class="fu">append</span>(df_list, <span class="fu">list</span>(count_data))</span>
<span id="cb31-39"><a href="server.html#cb31-39" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb31-40"><a href="server.html#cb31-40" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb31-41"><a href="server.html#cb31-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-42"><a href="server.html#cb31-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># generate a data frame containing count data from all experiments </span></span>
<span id="cb31-43"><a href="server.html#cb31-43" aria-hidden="true" tabindex="-1"></a>  final_df <span class="ot">=</span> <span class="fu">generate_combine_dataset</span>(df_list, genes, input<span class="sc">$</span>sex, col_list[selected_data], des_df<span class="sc">$</span>pain[selected_data], des_df<span class="sc">$</span>Species[selected_data], </span>
<span id="cb31-44"><a href="server.html#cb31-44" aria-hidden="true" tabindex="-1"></a>                                      expression, Condition, Population, symbol, Dataset, Species)</span>
<span id="cb31-45"><a href="server.html#cb31-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-46"><a href="server.html#cb31-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-47"><a href="server.html#cb31-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># plot the combined gene expression dot plot </span></span>
<span id="cb31-48"><a href="server.html#cb31-48" aria-hidden="true" tabindex="-1"></a>  combine_dot <span class="ot">=</span> <span class="fu">plotcombine_server</span>(<span class="st">&quot;dot&quot;</span>, final_df, input<span class="sc">$</span>sex, genes, Population, symbol, expression, <span class="st">&quot;Dataset&quot;</span>) <span class="co">#dotplot</span></span>
<span id="cb31-49"><a href="server.html#cb31-49" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(variables<span class="sc">$</span>pbmc) <span class="sc">!=</span> <span class="cn">TRUE</span>){</span>
<span id="cb31-50"><a href="server.html#cb31-50" aria-hidden="true" tabindex="-1"></a>    scrna_dot <span class="ot">=</span> <span class="fu">plothomescdot_server</span>(<span class="st">&quot;homespat&quot;</span>, variables<span class="sc">$</span>pbmc, genes)</span>
<span id="cb31-51"><a href="server.html#cb31-51" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb31-52"><a href="server.html#cb31-52" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-53"><a href="server.html#cb31-53" aria-hidden="true" tabindex="-1"></a>  <span class="co"># query differential expression (DE) data frames for plotting</span></span>
<span id="cb31-54"><a href="server.html#cb31-54" aria-hidden="true" tabindex="-1"></a>  degm <span class="ot">=</span> RSQLite<span class="sc">::</span><span class="fu">dbGetPreparedQuery</span>(conn, <span class="st">&quot;SELECT * FROM mouse_all_deg_df WHERE symbol = ?&quot;</span>, <span class="at">bind.data=</span><span class="fu">data.frame</span>(<span class="at">symbol=</span>genes))</span>
<span id="cb31-55"><a href="server.html#cb31-55" aria-hidden="true" tabindex="-1"></a>  sql  <span class="ot">=</span> <span class="st">&quot;SELECT * FROM rat_gene_data WHERE mgi_symbol = ?&quot;</span></span>
<span id="cb31-56"><a href="server.html#cb31-56" aria-hidden="true" tabindex="-1"></a>  rg   <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">dbGetPreparedQuery</span>(conn, sql, <span class="at">bind.data=</span><span class="fu">data.frame</span>(<span class="at">mgi_symbol=</span> genes)))<span class="sc">$</span>rgd_symbol</span>
<span id="cb31-57"><a href="server.html#cb31-57" aria-hidden="true" tabindex="-1"></a>  degr <span class="ot">=</span> RSQLite<span class="sc">::</span><span class="fu">dbGetPreparedQuery</span>(conn, <span class="st">&quot;SELECT * FROM rat_deg_df WHERE symbol = ?&quot;</span>, <span class="at">bind.data=</span><span class="fu">data.frame</span>(<span class="at">symbol=</span>rg))</span>
<span id="cb31-58"><a href="server.html#cb31-58" aria-hidden="true" tabindex="-1"></a>  sql  <span class="ot">=</span> <span class="st">&quot;SELECT * FROM human_gene_data WHERE mgi_symbol = ?&quot;</span></span>
<span id="cb31-59"><a href="server.html#cb31-59" aria-hidden="true" tabindex="-1"></a>  hg   <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">dbGetPreparedQuery</span>(conn, sql, <span class="at">bind.data=</span><span class="fu">data.frame</span>(<span class="at">mgi_symbol=</span> genes)))<span class="sc">$</span>hgnc_symbol</span>
<span id="cb31-60"><a href="server.html#cb31-60" aria-hidden="true" tabindex="-1"></a>  degh <span class="ot">=</span> RSQLite<span class="sc">::</span><span class="fu">dbGetPreparedQuery</span>(conn, <span class="st">&quot;SELECT * FROM human_deg_df WHERE symbol = ?&quot;</span>, <span class="at">bind.data=</span><span class="fu">data.frame</span>(<span class="at">symbol=</span>hg))</span>
<span id="cb31-61"><a href="server.html#cb31-61" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-62"><a href="server.html#cb31-62" aria-hidden="true" tabindex="-1"></a>  <span class="co"># generate combined DE data frames </span></span>
<span id="cb31-63"><a href="server.html#cb31-63" aria-hidden="true" tabindex="-1"></a>  combine_degplot <span class="ot">=</span> <span class="fu">deg_combine_server</span>(<span class="st">&quot;deg_plot&quot;</span>,degr, degm, degh, Population,symbol, log2FoldChange,sig, <span class="st">&quot;Dataset&quot;</span>)</span>
<span id="cb31-64"><a href="server.html#cb31-64" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-65"><a href="server.html#cb31-65" aria-hidden="true" tabindex="-1"></a>}) </span></code></pre></div>
</div>
<div id="individual-analysis-1" class="section level3 hasAnchor" number="4.2.4">
<h3><span class="header-section-number">4.2.4</span> Individual Analysis<a href="server.html#individual-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="data/Images/indiv.png" width="3453" /></p>
<p>First, a table that contains dataset description is rendered at the top.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="server.html#cb32-1" aria-hidden="true" tabindex="-1"></a>output<span class="sc">$</span>dataset_table <span class="ot">=</span> DT<span class="sc">::</span><span class="fu">renderDataTable</span>({</span>
<span id="cb32-2"><a href="server.html#cb32-2" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(</span>
<span id="cb32-3"><a href="server.html#cb32-3" aria-hidden="true" tabindex="-1"></a>  des_df[<span class="fu">c</span>(<span class="st">&quot;Dataset&quot;</span>, <span class="st">&quot;Pain_Model&quot;</span>, <span class="st">&quot;Species&quot;</span>, <span class="st">&quot;Tissue&quot;</span>)], <span class="do">## can select relevant columns</span></span>
<span id="cb32-4"><a href="server.html#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">6</span>,</span>
<span id="cb32-5"><a href="server.html#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">class =</span> <span class="st">&#39;nowrap&#39;</span>,</span>
<span id="cb32-6"><a href="server.html#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX =</span> <span class="cn">TRUE</span>, <span class="at">scrollY =</span> <span class="cn">TRUE</span>, <span class="at">pageLength =</span> <span class="dv">5</span>), <span class="at">selection =</span> <span class="st">&quot;single&quot;</span></span>
<span id="cb32-7"><a href="server.html#cb32-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb32-8"><a href="server.html#cb32-8" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<p>Next, the ui for each independent dataset is rendered in response to row selected from the table. This is passed by the <code>input$dataset_table_rows_selected</code> parameter and stored into a reactive variable indrows. The parameters required for calling the <code>shinypageUI</code> function is retrieved from the <code>des_df</code>.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="server.html#cb33-1" aria-hidden="true" tabindex="-1"></a>variables <span class="ot">=</span> <span class="fu">reactiveValues</span>({</span>
<span id="cb33-2"><a href="server.html#cb33-2" aria-hidden="true" tabindex="-1"></a>  indrows <span class="ot">=</span> <span class="cn">NULL</span> </span>
<span id="cb33-3"><a href="server.html#cb33-3" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb33-4"><a href="server.html#cb33-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-5"><a href="server.html#cb33-5" aria-hidden="true" tabindex="-1"></a>shiny<span class="sc">::</span><span class="fu">observe</span>({</span>
<span id="cb33-6"><a href="server.html#cb33-6" aria-hidden="true" tabindex="-1"></a>    variables<span class="sc">$</span>indrows <span class="ot">=</span>input<span class="sc">$</span>dataset_table_rows_selected</span>
<span id="cb33-7"><a href="server.html#cb33-7" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb33-8"><a href="server.html#cb33-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-9"><a href="server.html#cb33-9" aria-hidden="true" tabindex="-1"></a>output<span class="sc">$</span>shinypages  <span class="ot">=</span> shiny<span class="sc">::</span><span class="fu">renderUI</span>({</span>
<span id="cb33-10"><a href="server.html#cb33-10" aria-hidden="true" tabindex="-1"></a>    j <span class="ot">=</span> variables<span class="sc">$</span>indrows</span>
<span id="cb33-11"><a href="server.html#cb33-11" aria-hidden="true" tabindex="-1"></a>    select <span class="ot">=</span> des_df[j,]</span>
<span id="cb33-12"><a href="server.html#cb33-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb33-13"><a href="server.html#cb33-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if a scRNA sequencing data is selected </span></span>
<span id="cb33-14"><a href="server.html#cb33-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (select<span class="sc">$</span>type <span class="sc">==</span> <span class="st">&quot;single&quot;</span>){</span>
<span id="cb33-15"><a href="server.html#cb33-15" aria-hidden="true" tabindex="-1"></a>      <span class="fu">shinyscrnapageUI</span>(<span class="st">&quot;spatpage&quot;</span>, <span class="st">&quot;Human DRG spatial-seq&quot;</span>, <span class="at">des_dir=</span><span class="st">&quot;des/spatial.Rhtml&quot;</span>)</span>
<span id="cb33-16"><a href="server.html#cb33-16" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb33-17"><a href="server.html#cb33-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb33-18"><a href="server.html#cb33-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb33-19"><a href="server.html#cb33-19" aria-hidden="true" tabindex="-1"></a>      identifier <span class="ot">=</span> select<span class="sc">$</span>Name</span>
<span id="cb33-20"><a href="server.html#cb33-20" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb33-21"><a href="server.html#cb33-21" aria-hidden="true" tabindex="-1"></a>      <span class="co"># directory for description html files </span></span>
<span id="cb33-22"><a href="server.html#cb33-22" aria-hidden="true" tabindex="-1"></a>      des_dir <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;des/&quot;</span>,identifier, <span class="st">&quot;.Rhtml&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>) <span class="do">## you should put your description files here</span></span>
<span id="cb33-23"><a href="server.html#cb33-23" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb33-24"><a href="server.html#cb33-24" aria-hidden="true" tabindex="-1"></a>      <span class="co"># render ui </span></span>
<span id="cb33-25"><a href="server.html#cb33-25" aria-hidden="true" tabindex="-1"></a>      <span class="fu">shinypageUI</span>(name, select<span class="sc">$</span>Dataset,</span>
<span id="cb33-26"><a href="server.html#cb33-26" aria-hidden="true" tabindex="-1"></a>                  select<span class="sc">$</span>include_degs,</span>
<span id="cb33-27"><a href="server.html#cb33-27" aria-hidden="true" tabindex="-1"></a>                  select<span class="sc">$</span>include_subtypes, <span class="at">des_dir =</span> des_dir, <span class="at">includegoi =</span> select<span class="sc">$</span>include_gois, </span>
<span id="cb33-28"><a href="server.html#cb33-28" aria-hidden="true" tabindex="-1"></a>                  <span class="at">include_count =</span> select<span class="sc">$</span>include_count)</span>
<span id="cb33-29"><a href="server.html#cb33-29" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb33-30"><a href="server.html#cb33-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb33-31"><a href="server.html#cb33-31" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb33-32"><a href="server.html#cb33-32" aria-hidden="true" tabindex="-1"></a>  })</span></code></pre></div>
<p>The server functions for each study-specific ui called is shown below. The shinypage server module contains code for generating plots and result tables.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="server.html#cb34-1" aria-hidden="true" tabindex="-1"></a>shiny<span class="sc">::</span><span class="fu">observe</span>({</span>
<span id="cb34-2"><a href="server.html#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="server.html#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="co"># retrieve the row number selected </span></span>
<span id="cb34-4"><a href="server.html#cb34-4" aria-hidden="true" tabindex="-1"></a>i <span class="ot">=</span> input<span class="sc">$</span>dataset_table_rows_selected</span>
<span id="cb34-5"><a href="server.html#cb34-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-6"><a href="server.html#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">is.null</span>(i) <span class="sc">==</span> <span class="cn">FALSE</span>) {</span>
<span id="cb34-7"><a href="server.html#cb34-7" aria-hidden="true" tabindex="-1"></a>  selected <span class="ot">=</span> des_df[i,]</span>
<span id="cb34-8"><a href="server.html#cb34-8" aria-hidden="true" tabindex="-1"></a>  name <span class="ot">=</span> selected<span class="sc">$</span>Name</span>
<span id="cb34-9"><a href="server.html#cb34-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-10"><a href="server.html#cb34-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># the scRNA seq dataset requires different UI </span></span>
<span id="cb34-11"><a href="server.html#cb34-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (selected<span class="sc">$</span>Dataset <span class="sc">==</span> <span class="st">&quot;Human DRG spatial-seq&quot;</span>){</span>
<span id="cb34-12"><a href="server.html#cb34-12" aria-hidden="true" tabindex="-1"></a>    shiny<span class="sc">::</span><span class="fu">callModule</span>(shinyscrna, <span class="st">&quot;spatpage&quot;</span>, <span class="fu">reactive</span>({<span class="fu">genes</span>()}), variables<span class="sc">$</span>pbmc)</span>
<span id="cb34-13"><a href="server.html#cb34-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb34-14"><a href="server.html#cb34-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-15"><a href="server.html#cb34-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># for bulk RNA seq datasets</span></span>
<span id="cb34-16"><a href="server.html#cb34-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> {</span>
<span id="cb34-17"><a href="server.html#cb34-17" aria-hidden="true" tabindex="-1"></a>    shiny<span class="sc">::</span><span class="fu">callModule</span>(shinypage, name, input<span class="sc">$</span>sex, selected<span class="sc">$</span>count, <span class="fu">as.data.frame</span>(col_list[i]),</span>
<span id="cb34-18"><a href="server.html#cb34-18" aria-hidden="true" tabindex="-1"></a>                      selected<span class="sc">$</span>Species, selected<span class="sc">$</span>pain, <span class="fu">reactive</span>({<span class="fu">genes</span>()}), <span class="at">include_deg =</span> selected<span class="sc">$</span>include_degs, <span class="at">deg_df_name =</span> selected<span class="sc">$</span>deg_df_name,</span>
<span id="cb34-19"><a href="server.html#cb34-19" aria-hidden="true" tabindex="-1"></a>                      <span class="at">dataset =</span> selected<span class="sc">$</span>Dataset, <span class="at">include_subtype =</span> selected<span class="sc">$</span>include_subtypes, <span class="at">include_count =</span> selected<span class="sc">$</span>include_count,<span class="at">parent =</span> session)</span>
<span id="cb34-20"><a href="server.html#cb34-20" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb34-21"><a href="server.html#cb34-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-22"><a href="server.html#cb34-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-23"><a href="server.html#cb34-23" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
</div>
</div>
<div id="network-analysis" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Network Analysis<a href="server.html#network-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Network integration can give context when working with large datasets, and provide a platform to integrate information from multiple sources.</p>
<p><img src="data/Images/network.png" width="3312" /></p>
<p>Here, networks are built using protein–protein interaction data taken from the STRING database. The network can be enriched in various ways. As an example, we have enriched the STRING networks in our example by:</p>
<ol style="list-style-type: decimal">
<li>An <em>Enrichment Score</em>, an integrated statistic using data from all experiments.</li>
<li>Differential expression in an individual experiment.</li>
<li>Genes of interest, as a list</li>
</ol>
<ul>
<li>Here, we used known pain-associated genes from various sources</li>
</ul>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="server.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># update choice</span></span>
<span id="cb35-2"><a href="server.html#cb35-2" aria-hidden="true" tabindex="-1"></a>shiny<span class="sc">::</span><span class="fu">updateSelectizeInput</span>(session,</span>
<span id="cb35-3"><a href="server.html#cb35-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">inputId =</span> <span class="st">&quot;gene_symbols&quot;</span>,</span>
<span id="cb35-4"><a href="server.html#cb35-4" aria-hidden="true" tabindex="-1"></a>                            <span class="at">label =</span> <span class="st">&quot;Search Genes:&quot;</span>,</span>
<span id="cb35-5"><a href="server.html#cb35-5" aria-hidden="true" tabindex="-1"></a>                            <span class="at">choices =</span> human_gene_data<span class="sc">$</span>hgnc_symbol,</span>
<span id="cb35-6"><a href="server.html#cb35-6" aria-hidden="true" tabindex="-1"></a>                            <span class="at">server =</span> <span class="cn">TRUE</span>,</span>
<span id="cb35-7"><a href="server.html#cb35-7" aria-hidden="true" tabindex="-1"></a>                            <span class="at">selected =</span> <span class="fu">c</span>(<span class="st">&quot;ATF3&quot;</span>) <span class="do">## choose your favourite</span></span>
<span id="cb35-8"><a href="server.html#cb35-8" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<div id="integration-with-string-db" class="section level3 hasAnchor" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> Integration with STRING DB<a href="server.html#integration-with-string-db" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Node and edge data is extracted in real-time from STRING <a href="https://string-db.org/" class="uri">https://string-db.org/</a> and reconstructed into a network based on user input.</p>
<p><em>For reference only, you don’t need to modify this.</em></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="server.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="do">### this doesn&#39;t need to be modified</span></span>
<span id="cb36-2"><a href="server.html#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="co"># generates network  when the submit button is clicked </span></span>
<span id="cb36-3"><a href="server.html#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="fu">observeEvent</span>(input<span class="sc">$</span>submit, {</span>
<span id="cb36-4"><a href="server.html#cb36-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-5"><a href="server.html#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Query gene interactions </span></span>
<span id="cb36-6"><a href="server.html#cb36-6" aria-hidden="true" tabindex="-1"></a>  genes_list  <span class="ot">=</span> <span class="fu">as.vector</span>(input<span class="sc">$</span>gene_symbols)</span>
<span id="cb36-7"><a href="server.html#cb36-7" aria-hidden="true" tabindex="-1"></a>  genes       <span class="ot">=</span> <span class="fu">paste</span>(genes_list, <span class="at">collapse =</span> <span class="st">&quot;%0d&quot;</span>)</span>
<span id="cb36-8"><a href="server.html#cb36-8" aria-hidden="true" tabindex="-1"></a>  interactions <span class="ot">&lt;-</span> <span class="fu">query_interactions</span>(genes, <span class="dv">15</span>)</span>
<span id="cb36-9"><a href="server.html#cb36-9" aria-hidden="true" tabindex="-1"></a>  proteins    <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">c</span>(<span class="fu">unique</span>(interactions<span class="sc">$</span>preferredName_A), <span class="fu">unique</span>(interactions<span class="sc">$</span>preferredName_B)))</span>
<span id="cb36-10"><a href="server.html#cb36-10" aria-hidden="true" tabindex="-1"></a>  nodes       <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb36-11"><a href="server.html#cb36-11" aria-hidden="true" tabindex="-1"></a>  id          <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">c</span>(<span class="fu">unique</span>(interactions<span class="sc">$</span>stringId_A), <span class="fu">unique</span>(interactions<span class="sc">$</span>stringId_B)))</span>
<span id="cb36-12"><a href="server.html#cb36-12" aria-hidden="true" tabindex="-1"></a>  nodes       <span class="ot">=</span> <span class="fu">as.data.frame</span>(id)</span>
<span id="cb36-13"><a href="server.html#cb36-13" aria-hidden="true" tabindex="-1"></a>  nodes<span class="sc">$</span>label <span class="ot">=</span> proteins</span>
<span id="cb36-14"><a href="server.html#cb36-14" aria-hidden="true" tabindex="-1"></a>  nodes       <span class="ot">=</span> <span class="fu">mutate</span>(nodes, <span class="at">shape =</span> <span class="fu">ifelse</span>(nodes<span class="sc">$</span>label <span class="sc">%in%</span> input<span class="sc">$</span>gene_symbols, <span class="st">&quot;diamond&quot;</span>, <span class="st">&quot;circle&quot;</span>))</span>
<span id="cb36-15"><a href="server.html#cb36-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-16"><a href="server.html#cb36-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-17"><a href="server.html#cb36-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># annotate nodes by PES  </span></span>
<span id="cb36-18"><a href="server.html#cb36-18" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (input<span class="sc">$</span>pop <span class="sc">==</span> <span class="st">&quot;Enrichment Score&quot;</span>) {</span>
<span id="cb36-19"><a href="server.html#cb36-19" aria-hidden="true" tabindex="-1"></a>    mode <span class="ot">=</span> <span class="st">&quot;meta&quot;</span></span>
<span id="cb36-20"><a href="server.html#cb36-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-21"><a href="server.html#cb36-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># set metrics to PES </span></span>
<span id="cb36-22"><a href="server.html#cb36-22" aria-hidden="true" tabindex="-1"></a>    metrics <span class="ot">=</span> score_df</span>
<span id="cb36-23"><a href="server.html#cb36-23" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">=</span> score_df[score_df<span class="sc">$</span>symbol <span class="sc">%in%</span> proteins,]</span>
<span id="cb36-24"><a href="server.html#cb36-24" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-25"><a href="server.html#cb36-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># legends </span></span>
<span id="cb36-26"><a href="server.html#cb36-26" aria-hidden="true" tabindex="-1"></a>    output<span class="sc">$</span>netlegend <span class="ot">=</span> <span class="fu">renderUI</span>({</span>
<span id="cb36-27"><a href="server.html#cb36-27" aria-hidden="true" tabindex="-1"></a>      <span class="fu">HTML</span>(<span class="fu">paste0</span>(</span>
<span id="cb36-28"><a href="server.html#cb36-28" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;&lt;br&gt;&lt;br&gt;&quot;</span>,</span>
<span id="cb36-29"><a href="server.html#cb36-29" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;&lt;p style = &#39;display: block; margin-left: 20px; margin-right: auto;&#39;&gt;&amp;emsp;&amp;emsp;&amp;emsp;PES&lt;/p&gt;&quot;</span>,</span>
<span id="cb36-30"><a href="server.html#cb36-30" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;&lt;img style = &#39;display: block; margin-left: auto; margin-right: auto;&#39; src=&#39;bitmap.png&#39; width = &#39;80&#39; height = &#39;130&#39;&gt;&quot;</span>,<span class="st">&quot;&lt;br&gt;&quot;</span>)</span>
<span id="cb36-31"><a href="server.html#cb36-31" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb36-32"><a href="server.html#cb36-32" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb36-33"><a href="server.html#cb36-33" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb36-34"><a href="server.html#cb36-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-35"><a href="server.html#cb36-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># annotate by individual study </span></span>
<span id="cb36-36"><a href="server.html#cb36-36" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> {</span>
<span id="cb36-37"><a href="server.html#cb36-37" aria-hidden="true" tabindex="-1"></a>    mode <span class="ot">=</span> <span class="st">&quot;indiv&quot;</span></span>
<span id="cb36-38"><a href="server.html#cb36-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-39"><a href="server.html#cb36-39" aria-hidden="true" tabindex="-1"></a>    <span class="co"># &quot;data&quot; contains DEG data for all experiments </span></span>
<span id="cb36-40"><a href="server.html#cb36-40" aria-hidden="true" tabindex="-1"></a>    sql <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;SELECT * FROM &quot;</span>, input<span class="sc">$</span>pop, <span class="st">&quot;_deg_df&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb36-41"><a href="server.html#cb36-41" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-42"><a href="server.html#cb36-42" aria-hidden="true" tabindex="-1"></a>    <span class="co"># filter data by the selected experiment </span></span>
<span id="cb36-43"><a href="server.html#cb36-43" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">=</span> data[data<span class="sc">$</span>experiment_id <span class="sc">==</span> input<span class="sc">$</span>pop,]</span>
<span id="cb36-44"><a href="server.html#cb36-44" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">=</span> <span class="fu">mutate</span>(res, <span class="at">sig=</span><span class="fu">ifelse</span>((res<span class="sc">$</span>padj<span class="sc">&lt;</span><span class="fl">0.05</span>), <span class="st">&quot;SIG&quot;</span>, <span class="st">&quot;NS&quot;</span>))</span>
<span id="cb36-45"><a href="server.html#cb36-45" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-46"><a href="server.html#cb36-46" aria-hidden="true" tabindex="-1"></a>    <span class="co"># set metrics to DEG daata from a single study </span></span>
<span id="cb36-47"><a href="server.html#cb36-47" aria-hidden="true" tabindex="-1"></a>    metrics <span class="ot">=</span> res</span>
<span id="cb36-48"><a href="server.html#cb36-48" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">=</span> <span class="fu">filter</span>(res, symbol <span class="sc">%in%</span> proteins)  <span class="co"># this means the proteins must be in human symbol</span></span>
<span id="cb36-49"><a href="server.html#cb36-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(dt) <span class="ot">=</span> dt<span class="sc">$</span>symbol</span>
<span id="cb36-50"><a href="server.html#cb36-50" aria-hidden="true" tabindex="-1"></a>    dt <span class="ot">=</span> dt[<span class="fu">c</span>(<span class="st">&quot;log2FoldChange&quot;</span>,<span class="st">&quot;padj&quot;</span>, <span class="st">&quot;sig&quot;</span>)]</span>
<span id="cb36-51"><a href="server.html#cb36-51" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-52"><a href="server.html#cb36-52" aria-hidden="true" tabindex="-1"></a>    <span class="co"># display legend for deg  </span></span>
<span id="cb36-53"><a href="server.html#cb36-53" aria-hidden="true" tabindex="-1"></a>    output<span class="sc">$</span>netlegend <span class="ot">=</span> <span class="fu">renderUI</span>({</span>
<span id="cb36-54"><a href="server.html#cb36-54" aria-hidden="true" tabindex="-1"></a>      <span class="fu">HTML</span>(<span class="fu">paste0</span>(</span>
<span id="cb36-55"><a href="server.html#cb36-55" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;&lt;br&gt;&lt;br&gt;&quot;</span>,<span class="st">&quot;&lt;p style = &#39;display: block; margin-left: 20px; margin-right: auto;&#39;&gt;&amp;emsp;&amp;emsp;&amp;emsp;LFC&lt;/p&gt;</span></span>
<span id="cb36-56"><a href="server.html#cb36-56" aria-hidden="true" tabindex="-1"></a><span class="st">                      &lt;img style = &#39;display: block; margin-left: auto; margin-right: auto;&#39; src=&#39;legend.png&#39; width = &#39;80&#39; height = &#39;130&#39;&gt;</span></span>
<span id="cb36-57"><a href="server.html#cb36-57" aria-hidden="true" tabindex="-1"></a><span class="st">                      &quot;</span>,<span class="st">&quot;&lt;br&gt;&quot;</span>)</span>
<span id="cb36-58"><a href="server.html#cb36-58" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb36-59"><a href="server.html#cb36-59" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb36-60"><a href="server.html#cb36-60" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb36-61"><a href="server.html#cb36-61" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-62"><a href="server.html#cb36-62" aria-hidden="true" tabindex="-1"></a>  <span class="co"># node construction using get_node() function</span></span>
<span id="cb36-63"><a href="server.html#cb36-63" aria-hidden="true" tabindex="-1"></a>  nodes <span class="ot">=</span> <span class="fu">get_nodes</span>(interactions, gene_list, metrics, <span class="at">mode=</span>mode)</span>
<span id="cb36-64"><a href="server.html#cb36-64" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-65"><a href="server.html#cb36-65" aria-hidden="true" tabindex="-1"></a>  <span class="co"># construct edges </span></span>
<span id="cb36-66"><a href="server.html#cb36-66" aria-hidden="true" tabindex="-1"></a>  edges <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">from =</span> interactions<span class="sc">$</span>stringId_A, <span class="at">to =</span> interactions<span class="sc">$</span>stringId_B, <span class="at">width =</span> interactions<span class="sc">$</span>score,</span>
<span id="cb36-67"><a href="server.html#cb36-67" aria-hidden="true" tabindex="-1"></a>                      <span class="at">color =</span> <span class="st">&quot;#444444&quot;</span>)</span>
<span id="cb36-68"><a href="server.html#cb36-68" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-69"><a href="server.html#cb36-69" aria-hidden="true" tabindex="-1"></a>  <span class="co"># network visualisation</span></span>
<span id="cb36-70"><a href="server.html#cb36-70" aria-hidden="true" tabindex="-1"></a>  nw <span class="ot">=</span> visNetwork<span class="sc">::</span><span class="fu">visNetwork</span>(nodes, edges, <span class="at">height =</span> <span class="st">&quot;500px&quot;</span>, <span class="at">width =</span> <span class="st">&quot;100%&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-71"><a href="server.html#cb36-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">visPhysics</span>(<span class="at">enabled =</span> <span class="cn">FALSE</span>,<span class="at">solver =</span> <span class="st">&quot;forceAtlas2Based&quot;</span>,</span>
<span id="cb36-72"><a href="server.html#cb36-72" aria-hidden="true" tabindex="-1"></a>               <span class="at">forceAtlas2Based =</span> <span class="fu">list</span>(<span class="at">avoidOverlap =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-73"><a href="server.html#cb36-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">visEdges</span>(<span class="at">smooth =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-74"><a href="server.html#cb36-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">visInteraction</span>(<span class="at">navigationButtons =</span> <span class="cn">TRUE</span>, <span class="at">zoomSpeed =</span> <span class="fl">0.6</span>)</span>
<span id="cb36-75"><a href="server.html#cb36-75" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-76"><a href="server.html#cb36-76" aria-hidden="true" tabindex="-1"></a>  output<span class="sc">$</span>network <span class="ot">&lt;-</span> <span class="fu">renderVisNetwork</span>({nw})</span>
<span id="cb36-77"><a href="server.html#cb36-77" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-78"><a href="server.html#cb36-78" aria-hidden="true" tabindex="-1"></a>  <span class="co"># score metrics table </span></span>
<span id="cb36-79"><a href="server.html#cb36-79" aria-hidden="true" tabindex="-1"></a>  output<span class="sc">$</span>contrast_table <span class="ot">&lt;-</span> DT<span class="sc">::</span><span class="fu">renderDataTable</span>({</span>
<span id="cb36-80"><a href="server.html#cb36-80" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>(</span>
<span id="cb36-81"><a href="server.html#cb36-81" aria-hidden="true" tabindex="-1"></a>      dt,</span>
<span id="cb36-82"><a href="server.html#cb36-82" aria-hidden="true" tabindex="-1"></a>      <span class="at">width =</span> <span class="dv">6</span>,</span>
<span id="cb36-83"><a href="server.html#cb36-83" aria-hidden="true" tabindex="-1"></a>      <span class="at">class =</span> <span class="st">&#39;nowrap&#39;</span>,</span>
<span id="cb36-84"><a href="server.html#cb36-84" aria-hidden="true" tabindex="-1"></a>      <span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX =</span> <span class="cn">TRUE</span>, <span class="at">pageLength =</span> <span class="dv">8</span>)</span>
<span id="cb36-85"><a href="server.html#cb36-85" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb36-86"><a href="server.html#cb36-86" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb36-87"><a href="server.html#cb36-87" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-88"><a href="server.html#cb36-88" aria-hidden="true" tabindex="-1"></a>  <span class="co"># a table containing description and information of nodes(proteins) in the PPI network </span></span>
<span id="cb36-89"><a href="server.html#cb36-89" aria-hidden="true" tabindex="-1"></a>  output<span class="sc">$</span>protein_table <span class="ot">&lt;-</span> DT<span class="sc">::</span><span class="fu">renderDataTable</span>({</span>
<span id="cb36-90"><a href="server.html#cb36-90" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>(</span>
<span id="cb36-91"><a href="server.html#cb36-91" aria-hidden="true" tabindex="-1"></a>      <span class="fu">query_information</span>(proteins),</span>
<span id="cb36-92"><a href="server.html#cb36-92" aria-hidden="true" tabindex="-1"></a>      <span class="at">width =</span> <span class="dv">6</span>,</span>
<span id="cb36-93"><a href="server.html#cb36-93" aria-hidden="true" tabindex="-1"></a>      <span class="at">class =</span> <span class="st">&quot;wrap&quot;</span>,</span>
<span id="cb36-94"><a href="server.html#cb36-94" aria-hidden="true" tabindex="-1"></a>      <span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX =</span> <span class="cn">TRUE</span>, <span class="at">pageLength =</span> <span class="dv">8</span>)</span>
<span id="cb36-95"><a href="server.html#cb36-95" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb36-96"><a href="server.html#cb36-96" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb36-97"><a href="server.html#cb36-97" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-98"><a href="server.html#cb36-98" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
</div>
</div>
<div id="functions" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Functions<a href="server.html#functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><em>An run-down of functions used throughout the app. For basic applications these don’t need to be modified.</em> A function can be called and reused multiple times by passing parameters. For this example database, functions are written for rendering plots. To illustrate them, we will use an example count dataset.</p>
<div id="data-processing" class="section level3 hasAnchor" number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> Data processing<a href="server.html#data-processing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="server.html#cb37-1" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Atf3&quot;</span>, <span class="st">&quot;Msh6&quot;</span>, <span class="st">&quot;Dlst&quot;</span>, <span class="st">&quot;Slc2a9&quot;</span>)</span>
<span id="cb37-2"><a href="server.html#cb37-2" aria-hidden="true" tabindex="-1"></a>sex <span class="ot">=</span> <span class="st">&quot;Both&quot;</span></span>
<span id="cb37-3"><a href="server.html#cb37-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-4"><a href="server.html#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="co"># load data</span></span>
<span id="cb37-5"><a href="server.html#cb37-5" aria-hidden="true" tabindex="-1"></a>count   <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;./data/CountTables/TPM_mouse.csv&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb37-6"><a href="server.html#cb37-6" aria-hidden="true" tabindex="-1"></a>colData <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(<span class="st">&quot;./data/colData/&quot;</span>, <span class="st">&quot;TPM_mouse_colData.csv&quot;</span>), <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb37-7"><a href="server.html#cb37-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(colData) <span class="ot">=</span> colData<span class="sc">$</span>X</span>
<span id="cb37-8"><a href="server.html#cb37-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-9"><a href="server.html#cb37-9" aria-hidden="true" tabindex="-1"></a><span class="co"># filter data</span></span>
<span id="cb37-10"><a href="server.html#cb37-10" aria-hidden="true" tabindex="-1"></a>filt <span class="ot">=</span> count[count<span class="sc">$</span>symbol <span class="sc">%in%</span> genes,]</span>
<span id="cb37-11"><a href="server.html#cb37-11" aria-hidden="true" tabindex="-1"></a>matfilt <span class="ot">=</span> filt[,<span class="sc">!</span>(<span class="fu">names</span>(filt) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;symbol&quot;</span>))] </span>
<span id="cb37-12"><a href="server.html#cb37-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-13"><a href="server.html#cb37-13" aria-hidden="true" tabindex="-1"></a><span class="co"># rename the first column </span></span>
<span id="cb37-14"><a href="server.html#cb37-14" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(matfilt) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;gene&quot;</span>, <span class="fu">colnames</span>(matfilt)[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(matfilt))])</span>
<span id="cb37-15"><a href="server.html#cb37-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(matfilt) <span class="ot">=</span> matfilt<span class="sc">$</span>gene</span>
<span id="cb37-16"><a href="server.html#cb37-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-17"><a href="server.html#cb37-17" aria-hidden="true" tabindex="-1"></a><span class="co"># group based on experimental conditions </span></span>
<span id="cb37-18"><a href="server.html#cb37-18" aria-hidden="true" tabindex="-1"></a>tcounts <span class="ot">&lt;-</span> <span class="fu">t</span>(matfilt) <span class="sc">%&gt;%</span></span>
<span id="cb37-19"><a href="server.html#cb37-19" aria-hidden="true" tabindex="-1"></a>  base<span class="sc">::</span><span class="fu">merge</span>(colData, ., <span class="at">by=</span><span class="st">&quot;row.names&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb37-20"><a href="server.html#cb37-20" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">gather</span>(gene, expression, (<span class="fu">ncol</span>(.)<span class="sc">-</span><span class="fu">nrow</span>(matfilt)<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span>(<span class="fu">ncol</span>(.)))</span>
<span id="cb37-21"><a href="server.html#cb37-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-22"><a href="server.html#cb37-22" aria-hidden="true" tabindex="-1"></a>tcounts<span class="sc">$</span>symbol <span class="ot">=</span> filt[filt<span class="sc">$</span>X <span class="sc">%in%</span> tcounts<span class="sc">$</span>gene,]<span class="sc">$</span>symbol</span>
<span id="cb37-23"><a href="server.html#cb37-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-24"><a href="server.html#cb37-24" aria-hidden="true" tabindex="-1"></a><span class="co"># get median expression</span></span>
<span id="cb37-25"><a href="server.html#cb37-25" aria-hidden="true" tabindex="-1"></a>tcounts_med <span class="ot">=</span> <span class="fu">get_median</span>(tcounts, sex, expression, Condition, Population, symbol, Dataset, Species)</span>
<span id="cb37-26"><a href="server.html#cb37-26" aria-hidden="true" tabindex="-1"></a>tcounts_med <span class="ot">&lt;-</span> tcounts_med[<span class="sc">!</span>tcounts_med<span class="sc">$</span>Condition <span class="sc">%in%</span> <span class="st">&quot;SNI&quot;</span>, ] <span class="co"># remove pain samples</span></span></code></pre></div>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="server.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39; a function that gets the median expression of a count data </span></span>
<span id="cb38-2"><a href="server.html#cb38-2" aria-hidden="true" tabindex="-1"></a>get_median <span class="ot">&lt;-</span> <span class="cf">function</span>(df, sex, var, ...){</span>
<span id="cb38-3"><a href="server.html#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ifelse</span>(sex<span class="sc">==</span><span class="st">&quot;Both&quot;</span>, data <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">group_by</span>(...) <span class="sc">%&gt;%</span></span>
<span id="cb38-4"><a href="server.html#cb38-4" aria-hidden="true" tabindex="-1"></a>           dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">expression=</span><span class="fu">median</span>(<span class="fu">as.double</span>({{var}}))), df <span class="ot">&lt;-</span> tcounts <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">group_by</span>(..., Sex) <span class="sc">%&gt;%</span></span>
<span id="cb38-5"><a href="server.html#cb38-5" aria-hidden="true" tabindex="-1"></a>           dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">expression=</span><span class="fu">median</span>(<span class="fu">as.double</span>({{var}}))))</span>
<span id="cb38-6"><a href="server.html#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(data)</span>
<span id="cb38-7"><a href="server.html#cb38-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="dot-plots" class="section level3 hasAnchor" number="4.4.2">
<h3><span class="header-section-number">4.4.2</span> Dot plots<a href="server.html#dot-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>median counts <em>(here, transcripts per million (TPM))</em></li>
</ul>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="server.html#cb39-1" aria-hidden="true" tabindex="-1"></a>plot_dotplot <span class="ot">&lt;-</span> <span class="cf">function</span>(data, x_var, y_var, col_var, facet_var, sex){</span>
<span id="cb39-2"><a href="server.html#cb39-2" aria-hidden="true" tabindex="-1"></a>  x_var <span class="ot">=</span> <span class="fu">enquo</span>(x_var)</span>
<span id="cb39-3"><a href="server.html#cb39-3" aria-hidden="true" tabindex="-1"></a>  y_var <span class="ot">=</span> <span class="fu">enquo</span>(y_var)</span>
<span id="cb39-4"><a href="server.html#cb39-4" aria-hidden="true" tabindex="-1"></a>  col_var <span class="ot">=</span> <span class="fu">enquo</span>(col_var)</span>
<span id="cb39-5"><a href="server.html#cb39-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-6"><a href="server.html#cb39-6" aria-hidden="true" tabindex="-1"></a>  g <span class="ot">=</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="at">data =</span> data, <span class="fu">aes</span>(<span class="at">x=</span><span class="sc">!!</span>x_var, <span class="at">y =</span> <span class="sc">!!</span>y_var)) <span class="sc">+</span></span>
<span id="cb39-7"><a href="server.html#cb39-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_colour_viridis_c</span>(<span class="at">option =</span> <span class="st">&quot;magma&quot;</span>, <span class="at">end =</span> .<span class="dv">90</span>) <span class="sc">+</span></span>
<span id="cb39-8"><a href="server.html#cb39-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">col=</span><span class="fu">log</span>(<span class="sc">!!</span>col_var), <span class="at">size=</span><span class="fu">log</span>(<span class="sc">!!</span>col_var))) <span class="sc">+</span> </span>
<span id="cb39-9"><a href="server.html#cb39-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(.<span class="sc">~</span>.data[[facet_var]], <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space=</span><span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb39-10"><a href="server.html#cb39-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span>population_labels) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">col=</span><span class="st">&quot;log(TPM)&quot;</span>, <span class="at">size =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb39-11"><a href="server.html#cb39-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">size =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb39-12"><a href="server.html#cb39-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-13"><a href="server.html#cb39-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add themes </span></span>
<span id="cb39-14"><a href="server.html#cb39-14" aria-hidden="true" tabindex="-1"></a>  g <span class="ot">=</span> g <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb39-15"><a href="server.html#cb39-15" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb39-16"><a href="server.html#cb39-16" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust=</span> <span class="dv">1</span>),</span>
<span id="cb39-17"><a href="server.html#cb39-17" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb39-18"><a href="server.html#cb39-18" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb39-19"><a href="server.html#cb39-19" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(), <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.3</span>),</span>
<span id="cb39-20"><a href="server.html#cb39-20" aria-hidden="true" tabindex="-1"></a>                        <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>, <span class="at">margin =</span> <span class="fu">margin</span>(<span class="dv">6</span>, <span class="dv">2</span>, <span class="dv">6</span>, <span class="dv">2</span>)), </span>
<span id="cb39-21"><a href="server.html#cb39-21" aria-hidden="true" tabindex="-1"></a>                        <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.25</span>, <span class="st">&quot;mm&quot;</span>)) </span>
<span id="cb39-22"><a href="server.html#cb39-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-23"><a href="server.html#cb39-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-24"><a href="server.html#cb39-24" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (sex <span class="sc">==</span> <span class="st">&quot;Separate&quot;</span>) {</span>
<span id="cb39-25"><a href="server.html#cb39-25" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">=</span> g <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>Sex, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">Sex =</span> sexlabels), <span class="at">scales =</span><span class="st">&quot;free_x&quot;</span>)</span>
<span id="cb39-26"><a href="server.html#cb39-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb39-27"><a href="server.html#cb39-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(g)</span>
<span id="cb39-28"><a href="server.html#cb39-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb39-29"><a href="server.html#cb39-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-30"><a href="server.html#cb39-30" aria-hidden="true" tabindex="-1"></a>dotplot <span class="ot">=</span> <span class="fu">plot_dotplot</span>(tcounts_med, Population, symbol, expression, <span class="st">&quot;Dataset&quot;</span>, sex)</span>
<span id="cb39-31"><a href="server.html#cb39-31" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(dotplot)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-53-1.png" width="672" /></p>
</div>
<div id="line-plots" class="section level3 hasAnchor" number="4.4.3">
<h3><span class="header-section-number">4.4.3</span> Line plots<a href="server.html#line-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Median gene expression in case and control conditions</li>
<li>Linetype is encoded by the timepoint of treatment group (if exists)</li>
</ul>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="server.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39; @param group_vars symbol, Timepoint</span></span>
<span id="cb40-2"><a href="server.html#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39; @param linetype_var Timepoint</span></span>
<span id="cb40-3"><a href="server.html#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;</span></span>
<span id="cb40-4"><a href="server.html#cb40-4" aria-hidden="true" tabindex="-1"></a>plot_lineplot <span class="ot">&lt;-</span> <span class="cf">function</span>(data,sex, x_var, y_var, col_var, linetype_var, ...){</span>
<span id="cb40-5"><a href="server.html#cb40-5" aria-hidden="true" tabindex="-1"></a>  x_var <span class="ot">=</span> <span class="fu">enquo</span>(x_var)</span>
<span id="cb40-6"><a href="server.html#cb40-6" aria-hidden="true" tabindex="-1"></a>  y_var <span class="ot">=</span> <span class="fu">enquo</span>(y_var)</span>
<span id="cb40-7"><a href="server.html#cb40-7" aria-hidden="true" tabindex="-1"></a>  col_var <span class="ot">=</span> <span class="fu">enquo</span>(col_var)</span>
<span id="cb40-8"><a href="server.html#cb40-8" aria-hidden="true" tabindex="-1"></a>  group_vars <span class="ot">=</span> <span class="fu">enquos</span>(...)</span>
<span id="cb40-9"><a href="server.html#cb40-9" aria-hidden="true" tabindex="-1"></a>  linetype_var <span class="ot">=</span> <span class="fu">enquo</span>(linetype_var)</span>
<span id="cb40-10"><a href="server.html#cb40-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-11"><a href="server.html#cb40-11" aria-hidden="true" tabindex="-1"></a>  theme_line <span class="ot">=</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb40-12"><a href="server.html#cb40-12" aria-hidden="true" tabindex="-1"></a>                <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-13"><a href="server.html#cb40-13" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>),</span>
<span id="cb40-14"><a href="server.html#cb40-14" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>, <span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust=</span> <span class="dv">1</span>),</span>
<span id="cb40-15"><a href="server.html#cb40-15" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb40-16"><a href="server.html#cb40-16" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-17"><a href="server.html#cb40-17" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(), <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.3</span>))</span>
<span id="cb40-18"><a href="server.html#cb40-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-19"><a href="server.html#cb40-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-20"><a href="server.html#cb40-20" aria-hidden="true" tabindex="-1"></a>  lineplot <span class="ot">=</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x=</span><span class="sc">!!</span>x_var, <span class="at">y=</span><span class="sc">!!</span>y_var, <span class="at">group=</span><span class="fu">interaction</span>(<span class="sc">!!!</span>group_vars))) <span class="sc">+</span></span>
<span id="cb40-21"><a href="server.html#cb40-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_colour_viridis</span>(<span class="at">discrete=</span><span class="cn">TRUE</span>, <span class="at">end =</span> .<span class="dv">80</span>) <span class="sc">+</span></span>
<span id="cb40-22"><a href="server.html#cb40-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">color=</span><span class="sc">!!</span>col_var, <span class="at">linetype=</span><span class="sc">!!</span>linetype_var)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">col=</span><span class="sc">!!</span>col_var)) <span class="sc">+</span> theme_line <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Expression&quot;</span>) <span class="sc">+</span></span>
<span id="cb40-23"><a href="server.html#cb40-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">linetype=</span><span class="fu">guide_legend</span>(<span class="at">label.hjust=</span><span class="fl">0.5</span>))</span>
<span id="cb40-24"><a href="server.html#cb40-24" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (sex <span class="sc">==</span><span class="st">&#39;Separate&#39;</span>){</span>
<span id="cb40-25"><a href="server.html#cb40-25" aria-hidden="true" tabindex="-1"></a>    lineplot <span class="ot">=</span> lineplot <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>Sex, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">Sex =</span> sexlabels), <span class="at">scales =</span> <span class="st">&quot;free_x&quot;</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">col=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb40-26"><a href="server.html#cb40-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-27"><a href="server.html#cb40-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-28"><a href="server.html#cb40-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(lineplot)</span>
<span id="cb40-29"><a href="server.html#cb40-29" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-30"><a href="server.html#cb40-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-31"><a href="server.html#cb40-31" aria-hidden="true" tabindex="-1"></a>tcounts_m <span class="ot">=</span> <span class="fu">get_median</span>(tcounts, sex, <span class="at">var =</span> expression, Condition, symbol, Timepoint, Dataset, Species)</span>
<span id="cb40-32"><a href="server.html#cb40-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-33"><a href="server.html#cb40-33" aria-hidden="true" tabindex="-1"></a>lineplot <span class="ot">=</span> <span class="fu">plot_lineplot</span>(tcounts_m, sex, Condition, expression, symbol, Timepoint, symbol, Timepoint)</span>
<span id="cb40-34"><a href="server.html#cb40-34" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(lineplot) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-54-1.png" width="672" /></p>
</div>
<div id="a-differential-expression-dot-plot" class="section level3 hasAnchor" number="4.4.4">
<h3><span class="header-section-number">4.4.4</span> A differential expression dot plot<a href="server.html#a-differential-expression-dot-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Colour intensity encodes the log2 fold change</li>
<li>Dot represents if the change is significant (filled) or not (unfilled).</li>
</ul>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="server.html#cb41-1" aria-hidden="true" tabindex="-1"></a>plot_degplot <span class="ot">&lt;-</span> <span class="cf">function</span>(data, x_var, y_var, col_var, shape_var){</span>
<span id="cb41-2"><a href="server.html#cb41-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb41-3"><a href="server.html#cb41-3" aria-hidden="true" tabindex="-1"></a>  x_var <span class="ot">=</span> <span class="fu">enquo</span>(x_var)</span>
<span id="cb41-4"><a href="server.html#cb41-4" aria-hidden="true" tabindex="-1"></a>  y_var <span class="ot">=</span> <span class="fu">enquo</span>(y_var)</span>
<span id="cb41-5"><a href="server.html#cb41-5" aria-hidden="true" tabindex="-1"></a>  col_var <span class="ot">=</span> <span class="fu">enquo</span>(col_var)</span>
<span id="cb41-6"><a href="server.html#cb41-6" aria-hidden="true" tabindex="-1"></a>  shape_var <span class="ot">=</span> <span class="fu">enquo</span>(shape_var)</span>
<span id="cb41-7"><a href="server.html#cb41-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb41-8"><a href="server.html#cb41-8" aria-hidden="true" tabindex="-1"></a>  degplot <span class="ot">=</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">interaction</span>(<span class="sc">!!</span>x_var), <span class="at">y =</span> <span class="sc">!!</span>y_var,</span>
<span id="cb41-9"><a href="server.html#cb41-9" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">text =</span> <span class="fu">paste</span>(<span class="st">&#39;padj: &#39;</span>,padj))) <span class="sc">+</span></span>
<span id="cb41-10"><a href="server.html#cb41-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_colour_viridis_c</span>(<span class="at">option =</span> <span class="st">&quot;viridis&quot;</span>, <span class="at">end =</span> .<span class="dv">90</span>) <span class="sc">+</span></span>
<span id="cb41-11"><a href="server.html#cb41-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">col=</span><span class="sc">!!</span>col_var, <span class="at">shape=</span><span class="sc">!!</span>shape_var, <span class="at">size=</span><span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">scale_shape_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">19</span>)) <span class="sc">+</span></span>
<span id="cb41-12"><a href="server.html#cb41-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">shape =</span> <span class="st">&quot;&quot;</span>, <span class="at">size =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span> <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span>subpopulation_labels) <span class="sc">+</span> </span>
<span id="cb41-13"><a href="server.html#cb41-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">size =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb41-14"><a href="server.html#cb41-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb41-15"><a href="server.html#cb41-15" aria-hidden="true" tabindex="-1"></a>  degplot <span class="ot">=</span> degplot <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb41-16"><a href="server.html#cb41-16" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb41-17"><a href="server.html#cb41-17" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust=</span> <span class="dv">1</span>),</span>
<span id="cb41-18"><a href="server.html#cb41-18" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb41-19"><a href="server.html#cb41-19" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb41-20"><a href="server.html#cb41-20" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(), <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.3</span>),</span>
<span id="cb41-21"><a href="server.html#cb41-21" aria-hidden="true" tabindex="-1"></a>                        <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>, <span class="at">margin =</span> <span class="fu">margin</span>(<span class="dv">6</span>, <span class="dv">2</span>, <span class="dv">6</span>, <span class="dv">2</span>)), </span>
<span id="cb41-22"><a href="server.html#cb41-22" aria-hidden="true" tabindex="-1"></a>                        <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.25</span>, <span class="st">&quot;mm&quot;</span>)) </span>
<span id="cb41-23"><a href="server.html#cb41-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb41-24"><a href="server.html#cb41-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(degplot)</span>
<span id="cb41-25"><a href="server.html#cb41-25" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb41-26"><a href="server.html#cb41-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-27"><a href="server.html#cb41-27" aria-hidden="true" tabindex="-1"></a>degplot <span class="ot">=</span> <span class="fu">plot_degplot</span>(deg_df, Population, symbol, log2FoldChange, sig)</span>
<span id="cb41-28"><a href="server.html#cb41-28" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(degplot)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-56-1.png" width="672" /></p>
</div>
<div id="a-volcano-plot" class="section level3 hasAnchor" number="4.4.5">
<h3><span class="header-section-number">4.4.5</span> A volcano plot<a href="server.html#a-volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>showing case vs control samples</li>
<li>genes of interest are labelled using colour.</li>
</ul>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="server.html#cb42-1" aria-hidden="true" tabindex="-1"></a>plot_volcanoplot <span class="ot">&lt;-</span> <span class="cf">function</span>(data, il_genes, x_var, y_var, label_var){</span>
<span id="cb42-2"><a href="server.html#cb42-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb42-3"><a href="server.html#cb42-3" aria-hidden="true" tabindex="-1"></a>  x_var <span class="ot">=</span> <span class="fu">enquo</span>(x_var)</span>
<span id="cb42-4"><a href="server.html#cb42-4" aria-hidden="true" tabindex="-1"></a>  y_var <span class="ot">=</span> <span class="fu">enquo</span>(y_var)</span>
<span id="cb42-5"><a href="server.html#cb42-5" aria-hidden="true" tabindex="-1"></a>  label_var <span class="ot">=</span> <span class="fu">enquo</span>(label_var)</span>
<span id="cb42-6"><a href="server.html#cb42-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb42-7"><a href="server.html#cb42-7" aria-hidden="true" tabindex="-1"></a>  volcanoplot <span class="ot">=</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="at">data =</span> data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="sc">!!</span>x_var, <span class="at">y =</span> <span class="sc">!!</span>y_var)) <span class="sc">+</span></span>
<span id="cb42-8"><a href="server.html#cb42-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">colour =</span> <span class="st">&quot;grey&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb42-9"><a href="server.html#cb42-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> il_genes, <span class="co"># New layer containing data subset il_genes</span></span>
<span id="cb42-10"><a href="server.html#cb42-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">size =</span> <span class="dv">2</span>,</span>
<span id="cb42-11"><a href="server.html#cb42-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">shape =</span> <span class="dv">21</span>,</span>
<span id="cb42-12"><a href="server.html#cb42-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">&quot;firebrick&quot;</span>,</span>
<span id="cb42-13"><a href="server.html#cb42-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb42-14"><a href="server.html#cb42-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>),</span>
<span id="cb42-15"><a href="server.html#cb42-15" aria-hidden="true" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb42-16"><a href="server.html#cb42-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="fu">log2</span>(<span class="fl">0.5</span>), <span class="fu">log2</span>(<span class="dv">2</span>)),</span>
<span id="cb42-17"><a href="server.html#cb42-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb42-18"><a href="server.html#cb42-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_label_repel</span>(<span class="at">data =</span> il_genes,</span>
<span id="cb42-19"><a href="server.html#cb42-19" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">aes</span>(<span class="at">label =</span> <span class="sc">!!</span>label_var),</span>
<span id="cb42-20"><a href="server.html#cb42-20" aria-hidden="true" tabindex="-1"></a>                     <span class="at">force =</span> <span class="dv">2</span>,</span>
<span id="cb42-21"><a href="server.html#cb42-21" aria-hidden="true" tabindex="-1"></a>                     <span class="at">nudge_y =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb42-22"><a href="server.html#cb42-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># scale_colour_manual(values = cols) +</span></span>
<span id="cb42-23"><a href="server.html#cb42-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="fu">seq</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">2</span>)),</span>
<span id="cb42-24"><a href="server.html#cb42-24" aria-hidden="true" tabindex="-1"></a>                       <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">10</span>)) <span class="sc">+</span> </span>
<span id="cb42-25"><a href="server.html#cb42-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># theme_line +</span></span>
<span id="cb42-26"><a href="server.html#cb42-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;-log10(FDR)&quot;</span>)</span>
<span id="cb42-27"><a href="server.html#cb42-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb42-28"><a href="server.html#cb42-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(volcanoplot)</span>
<span id="cb42-29"><a href="server.html#cb42-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb42-30"><a href="server.html#cb42-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb42-31"><a href="server.html#cb42-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-32"><a href="server.html#cb42-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-33"><a href="server.html#cb42-33" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">mutate</span>(deg, <span class="at">log10fdr=</span><span class="sc">-</span><span class="fu">log10</span>(padj))</span>
<span id="cb42-34"><a href="server.html#cb42-34" aria-hidden="true" tabindex="-1"></a>volc_df <span class="ot">=</span> res[res<span class="sc">$</span>symbol <span class="sc">%in%</span> genes,]</span>
<span id="cb42-35"><a href="server.html#cb42-35" aria-hidden="true" tabindex="-1"></a>volcanoplot <span class="ot">=</span> <span class="fu">plot_volcanoplot</span>(res, volc_df, log2FoldChange, log10fdr, symbol)</span>
<span id="cb42-36"><a href="server.html#cb42-36" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(volcanoplot)</span></code></pre></div>
<pre><code>## Warning: Removed 31279 rows containing missing values (`geom_point()`).</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-57-1.png" width="672" /></p>
</div>
<div id="other" class="section level3 hasAnchor" number="4.4.6">
<h3><span class="header-section-number">4.4.6</span> Other<a href="server.html#other" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A downloadable R Markdown report can be generated by clicking the “Generate Code” button, as shown below.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="server.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># download a R markdown report </span></span>
<span id="cb44-2"><a href="server.html#cb44-2" aria-hidden="true" tabindex="-1"></a>output<span class="sc">$</span>combineplot <span class="ot">&lt;-</span> <span class="fu">downloadHandler</span>(</span>
<span id="cb44-3"><a href="server.html#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">filename =</span> <span class="st">&quot;RNAseq_homeplot.html&quot;</span>,</span>
<span id="cb44-4"><a href="server.html#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">content =</span> <span class="cf">function</span>(file) {</span>
<span id="cb44-5"><a href="server.html#cb44-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb44-6"><a href="server.html#cb44-6" aria-hidden="true" tabindex="-1"></a>    params <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">matrix =</span> final_df, <span class="at">genes =</span> genes, <span class="at">sex =</span> input<span class="sc">$</span>sex, <span class="at">scRNA =</span> variables<span class="sc">$</span>pbmc, <span class="at">ratdeg =</span> degr,</span>
<span id="cb44-7"><a href="server.html#cb44-7" aria-hidden="true" tabindex="-1"></a>                   <span class="at">mousedeg =</span> degm, <span class="at">humandeg =</span> degh,</span>
<span id="cb44-8"><a href="server.html#cb44-8" aria-hidden="true" tabindex="-1"></a>                   <span class="at">human_gene_data =</span> human_gene_data)</span>
<span id="cb44-9"><a href="server.html#cb44-9" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb44-10"><a href="server.html#cb44-10" aria-hidden="true" tabindex="-1"></a>    shiny<span class="sc">::</span><span class="fu">withProgress</span>(<span class="at">value =</span> <span class="dv">0</span>,</span>
<span id="cb44-11"><a href="server.html#cb44-11" aria-hidden="true" tabindex="-1"></a>                        <span class="at">message =</span> <span class="st">&#39;Rendering plotting report&#39;</span>,</span>
<span id="cb44-12"><a href="server.html#cb44-12" aria-hidden="true" tabindex="-1"></a>                        <span class="at">detail =</span>  <span class="st">&#39;This might take several minutes.&#39;</span>,{</span>
<span id="cb44-13"><a href="server.html#cb44-13" aria-hidden="true" tabindex="-1"></a>                          rmarkdown<span class="sc">::</span><span class="fu">render</span>(<span class="at">input =</span> <span class="st">&quot;reports/combineplot.Rmd&quot;</span>,</span>
<span id="cb44-14"><a href="server.html#cb44-14" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">params =</span> params,</span>
<span id="cb44-15"><a href="server.html#cb44-15" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">output_file =</span> file,</span>
<span id="cb44-16"><a href="server.html#cb44-16" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">envir =</span> <span class="fu">new.env</span>(<span class="at">parent =</span> <span class="fu">globalenv</span>()))</span>
<span id="cb44-17"><a href="server.html#cb44-17" aria-hidden="true" tabindex="-1"></a>                        })</span>
<span id="cb44-18"><a href="server.html#cb44-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb44-19"><a href="server.html#cb44-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb44-20"><a href="server.html#cb44-20" aria-hidden="true" tabindex="-1"></a>) <span class="co"># closure for download handler</span></span></code></pre></div>
<p>The plots can also be downloaded as a zip file via a download button “Download Data.”</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="server.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># download all plots into a zip</span></span>
<span id="cb45-2"><a href="server.html#cb45-2" aria-hidden="true" tabindex="-1"></a>output<span class="sc">$</span>combineplots <span class="ot">&lt;-</span> <span class="fu">downloadHandler</span>(</span>
<span id="cb45-3"><a href="server.html#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">filename =</span> <span class="cf">function</span>() {</span>
<span id="cb45-4"><a href="server.html#cb45-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste</span>(<span class="st">&quot;combined&quot;</span>, <span class="st">&quot;zip&quot;</span>, <span class="at">sep=</span><span class="st">&quot;.&quot;</span>)</span>
<span id="cb45-5"><a href="server.html#cb45-5" aria-hidden="true" tabindex="-1"></a>  },</span>
<span id="cb45-6"><a href="server.html#cb45-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb45-7"><a href="server.html#cb45-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">content =</span> <span class="cf">function</span>(file){</span>
<span id="cb45-8"><a href="server.html#cb45-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set temporary working directory</span></span>
<span id="cb45-9"><a href="server.html#cb45-9" aria-hidden="true" tabindex="-1"></a>    owd <span class="ot">&lt;-</span> <span class="fu">setwd</span>(<span class="fu">tempdir</span>())</span>
<span id="cb45-10"><a href="server.html#cb45-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">on.exit</span>(<span class="fu">setwd</span>(owd))</span>
<span id="cb45-11"><a href="server.html#cb45-11" aria-hidden="true" tabindex="-1"></a>    fs <span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb45-12"><a href="server.html#cb45-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-13"><a href="server.html#cb45-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save the plots</span></span>
<span id="cb45-14"><a href="server.html#cb45-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggsave</span>(<span class="st">&#39;combine_deg.png&#39;</span>, <span class="at">plot =</span> combine_degplot, <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;in&quot;</span>, <span class="at">device=</span><span class="st">&#39;png&#39;</span>)</span>
<span id="cb45-15"><a href="server.html#cb45-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggsave</span>(<span class="st">&#39;combine_dot.png&#39;</span>, <span class="at">plot =</span> combine_dot, <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;in&quot;</span>, <span class="at">device=</span><span class="st">&#39;png&#39;</span>)</span>
<span id="cb45-16"><a href="server.html#cb45-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggsave</span>(<span class="st">&#39;scrna_dot.png&#39;</span>, <span class="at">plot =</span> scrna_dot, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">8</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">units =</span> <span class="st">&quot;in&quot;</span>, <span class="at">device=</span><span class="st">&#39;png&#39;</span>)</span>
<span id="cb45-17"><a href="server.html#cb45-17" aria-hidden="true" tabindex="-1"></a>    fs <span class="ot">=</span> <span class="fu">c</span>(fs, <span class="st">&#39;combine_deg.png&#39;</span>)</span>
<span id="cb45-18"><a href="server.html#cb45-18" aria-hidden="true" tabindex="-1"></a>    fs <span class="ot">=</span> <span class="fu">c</span>(fs, <span class="st">&#39;combine_dot.png&#39;</span>)</span>
<span id="cb45-19"><a href="server.html#cb45-19" aria-hidden="true" tabindex="-1"></a>    fs <span class="ot">=</span> <span class="fu">c</span>(fs, <span class="st">&#39;scrna_dot.png&#39;</span>)</span>
<span id="cb45-20"><a href="server.html#cb45-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-21"><a href="server.html#cb45-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># csv files for gene count data</span></span>
<span id="cb45-22"><a href="server.html#cb45-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.csv</span>(final_df,<span class="st">&quot;result_table.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb45-23"><a href="server.html#cb45-23" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-24"><a href="server.html#cb45-24" aria-hidden="true" tabindex="-1"></a>    fs <span class="ot">=</span> <span class="fu">c</span>(fs, <span class="st">&quot;result_table.csv&quot;</span>)</span>
<span id="cb45-25"><a href="server.html#cb45-25" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-26"><a href="server.html#cb45-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">zip</span>(file, fs)</span>
<span id="cb45-27"><a href="server.html#cb45-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb45-28"><a href="server.html#cb45-28" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="ui.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="deploy.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
